---
layout: home
# articles:
#   excerpt_type: html
---

<!DOCTYPE html>
<html>
<head>
<meta name="viewport" content="width=device-width, initial-scale=1">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">
<style>


/* Fixed sidenav, full height */
.sidenav {
  height: 80%;
  width: 200px;
  position: fixed;
  z-index: auto;
  top: 20;
  left: 0;
  background-color: 00FFFFFF;
  overflow-x: hidden;
  padding-top: 20px;
}

/* Style the sidenav links and the dropdown button */
.sidenav a, .dropdown-btn {
  padding: 6px 8px 6px 16px;
  text-decoration: none;
  font-size: 14px;
  color: #818181;
  display: block;
  border: none;
  background: none;
  width: 100%;
  text-align: left;
  cursor: pointer;
  outline: none;
}

/* On mouse-over */
.sidenav a:hover, .dropdown-btn:hover {
  color: grey;
}

/* Main content */
.main {
  justify-content: center;
  align-items: center;
  font-size: 16px; /* Increased text to enable scrolling */
  padding: 0px 10px;
}

/* Add an active class to the active dropdown button */
.active {
  background-color: black;
  color: #f1f1f1;
}

/* Dropdown container (hidden by default). Optional: add a lighter background color and some left padding to change the 
design of the dropdown content */
.dropdown-container {
  display: none;
  background-color: 00FFFFFF;
  padding-left: 10px;
}

/* Optional: Style the caret down icon */
.fa-caret-down {
  float: right;
  padding-right: 8px;
}

/* Some media queries for responsiveness */
@media screen and (max-height: 450px) {
  .sidenav {padding-top: 15px;}
  .sidenav a {font-size: 14px;}
}
</style>
</head>
<body>

<div class="sidenav">
  <a href="Bioinformatics_Workshop.html">Overview</a>
  <a href="selection.html">Participant Selection Process</a>
<button class="dropdown-btn">Partnering Institutions
    <i class="fa fa-caret-down"></i>
  </button>
  <div class="dropdown-container">
    <a href="crl.html">Central Research Laboratory,KIMS,India</a>
    <a href="uniandes.html">Universidad de los Andes,Bogotá</a>
    <a href="mrcgambia.html">The MRC Unit at LSHTM,The Gambia</a>
    <a href="pakistan.html">Lahore University of Management Sciences & Agha Khan University,Pakistan</a>
  </div>
  <button class="dropdown-btn">Course Content
    <i class="fa fa-caret-down"></i>
  </button>
  <div class="dropdown-container">
    <a href="computing_platforms.html">Data and Computational Platforms</a>
    <a href="#">Next Generation Sequencing</a>
    <a href="Linux.html">Introduction to Linux</a>
    <a href="ngs_file_formats.html">NGS File Formats</a>
    <a href="ngs_data_QC.html">NGS Data Quality Control</a>
    <a href="adapter_trimming.html">Adapter Trimming</a>
    <a href="taxonomy.html">Taxonomy Sequence Classification</a>
    <a href="denovo_assembly.html">de novo Assembly</a>
    <a href="quality_assessment_genome_assemblies.html">Quality Assessment 
of Genome Assemblies</a>
    <a href="genome_annotation.html">Genome Annotation</a>
    <a href="serotyping.html">Serotyping</a>
    <a href="mlst.html">Multilocus Sequence Typing</a>
    <a href="amr.html">AMR Profiling</a>
    <a href="#">Mapping</a>
    <a href="#">Variant Calling</a>
    <a href="#">Phylogenetics</a>
    <a href="#">Bioinformatics Pipelines</a>
    <a href="#">Case Studies</a>
 </div>
  <a href="#">Trainers</a>  
  <a href="#contact">Acknowledgement</a>
</div> 

<p align="justify">To analyse genomic data, you will need:</p>

<ol type="i"; style="padding-left: 30px">
<li>Genomic data</li>
<li>Robust computers</li>
<li>Bioinformatic tools</li>
<li>Electricity and internet connectivity</li>
</ol>
</p>


<h4>(i) Genomic data</h4>

<p align="justify">Data to learn this course can accessed using this link:</p>


<h4>(ii) Computers</h4>

<h5>(a) Definitions</h5>

<p align="justify">We will begin by defining a few concepts about computers that will be 
frequently used throughout this course.</p>

<ul align="justify"; style="padding-left: 30px">
<li><b>The Central Processing Unit (CPU)</b> - also known as the processor, provides the 
instructions and processing power the computer needs to do its work. It executes instructions of 
a computer program, such as arithmetic, logic, controlling, and input/output (I/O) operations. 
Within the CPU, we have cores and threads. They represent the CPU's processing capabilities and 
play a crucial role in its performance.
<ol style=“padding-left: 40px"><b>Cores:</b> Cores are individual processing units within a CPU. 
Each core is capable of executing instructions and performing calculations independently. A CPU 
with multiple cores can handle multiple tasks or threads simultaneously, which can lead to 
improved multitasking and overall performance. For example, a quad-core CPU has four cores, and 
an octa-core CPU has eight cores.</ol>
<ol style=“padding-left: 40px"><b>Threads:</b> Threads, in the context of a CPU, refer to virtual 
processing units. They are created by the CPU to handle specific tasks or processes. Each core 
can often support multiple threads, which allows the CPU to work on multiple tasks concurrently. 
This technology is known as multithreading. For instance, a CPU with hyper-threading technology 
might have two threads per core, effectively doubling the number of threads it can handle.</ol>

<p align=”justify”; style=“padding-left: 40px”><b><i>Note:</b>Having multiple cores and threads 
can 
significantly improve a CPU's performance, especially in scenarios where multiple tasks or 
applications are running simultaneously, such as in modern computing environments where 
multitasking is common. It enables smoother and more efficient processing of tasks, reducing 
bottlenecks and enhancing overall system responsiveness.</p></i>
</li>

<li><b>Random Access Memory (RAM) </b>  - it is your computer's short-term memory. This is where 
the data is stored that your computer processor needs to run your applications and open your 
files.</li>
<li><b>Hard disk drive (HDD) </b> - HDD Is a data storage device used in computers to store and 
retrieve digital information. It is a non-volatile storage medium, meaning it retains data even 
when the power is turned off. HDDs have been a common form of secondary storage for many years, 
although they are gradually being replaced by Solid-State Drives (SSDs) in many applications due 
to the latter's faster speed and other advantages.</li>
<li><b>Solid-state drive (SSD) space </b> - SSD is a storage device that uses fast and durable 
memory chips to store data in electronic form. It provides storage space for your files, 
applications, and operating system in your computer or device, serving as a modern and speedy 
alternative to traditional HDDs</li>
</ul>
</p>

<p align="center"><img width="600" height="240" alt="logo"
src="assets/images/logo/comp_1.png"></p>

<br>

<p align=”justify”>To successfully complete this course, you will need ~100GB HDD/SSD free space 
in a single partition/drive and atleast 8GB of RAM, to store input data (i.e., raw reads [fastq 
files]and databases [kraken - 8GB; serotyping databases (seroba_k71_14082017 & GBS-SBG); and 
popPUNK database - 21GB), and output data.</p>

<p align=”justify”>Outside this course, the choice of computing platform can be selected based on 
amount of data generated per year. For example, if generating:</p>

<ul align="justify"; style="padding-left: 30px">

<li><100 raw reads in a year, you can use a laptop/personal computer to analyse your data.</li>

<li>between 100 - 1000 reads, you might  need a workstation which is a high-performance computer 
system that is basically designed for a single user and has advanced graphic capabilities, large 
storage capacity, and a powerful central processing unit. A workstation is more capable than a 
personal computer(PC)/laptop but is less advanced than a server.</li>

<li>>1000 raw reads in a year, you will need server/cluster - a server is a specialised computer 
or software system designed to provide services, resources (especially CPU and RAM), or data to 
other computers or devices, known as clients, over a network. Clusters are groups of servers that 
are managed together and participate in workload management.</li>
</ul>

<p align="center"><img width="600" height="340" alt="logo"
src="assets/images/logo/comp_2.png"></p>

<h5>(b) Operating systems</h5>

<p align="justify">An operating system (OS) is a fundamental software component that 
manages and controls a computer's hardware resources and provides a foundation for 
running applications. It serves as an intermediary between the hardware and 
software, allowing users and applications to interact with and utilize the 
computer's resources effectively.</p>

<p align="justify">Unix, Linux, and Windows are three different families of 
OS used on computers and servers. Here are some key differences 
between them:

<ul align=”justify”; style="padding-left: 30px">
<li><b>Unix</b>: is one of the earliest operating systems, developed at AT&T's Bell 
Labs in the late 1960s. It has a rich history and served as the inspiration for many 
other operating systems. Unix operating systems are often proprietary and require 
licenses. Different flavors of Unix (e.g., AIX, HP-UX, Solaris) have different 
owners and licensing terms.</li>

<li><b>Linux</b>: is a Unix-like operating system that was developed in the 
early 1990s by Linus Torvalds. It is based on Unix principles and design but is 
open-source and community-driven. Linux is open-source and distributed under various 
licenses, such as the GNU General Public License (GPL). This means it is generally 
free to use, modify, and distribute. Types of Linux distributions include: 
Ubuntu, Debian, Fedora etc.</li>

<li><b>Windows</b>: is developed by Microsoft and has its roots in MS-DOS, 
which was released in the 1980s. Windows has a distinct development history from 
Unix-based systems.Windows is proprietary software, and users need to purchase 
licenses for its use. Microsoft offers different editions with varying features and 
licensing models.</li>

</ul> </p>

<p align="justify"; style="padding-left: 30px"><i><b>Note</b></i></p>
<p align="justify"; style="padding-left: 30px"><i>Both Unix and Linux provide 
text-based and graphical user interfaces. Users can choose to work in a 
command-line environment or use graphical desktop environments. While Unix and Linux 
have graphical interfaces, they are known for their powerful command-line 
interfaces. Many tasks and system administration can be performed more efficiently 
through the command line. These operating systems use a hierarchical file system 
with directories and files. File paths are case-sensitive.

<br>
<br>

Windows is primarily known for its graphical user interface (GUI), with versions 
like Windows 10 and 11 offering a modern, user-friendly desktop interface. Windows 
has a strong emphasis on GUI-based interaction, and many users may not need to use 
the command line for day-to-day tasks. </i></p>

<p align="justify"; style="padding-left: 30px"><i><b>If you are a Windows user, we 
recommend that you install Linux OS given that it has powerful command-line 
interfaces. </b></i></p>

<p align="justify"; style="padding-left:70px"><i><b>How to install Ubuntu in Windows 
computer</i></b>

<p align="justify"; style="padding-left:70px"><i>To install Ubuntu on a 
Windows computer, you can use the Windows Subsystem for Linux (WSL), which 
allows you to run a Linux distribution alongside your existing Windows 
installation without the need for dual-booting. Here are the steps to 
install Ubuntu using WSL:</i></p>

<p align="justify"; style="padding-left:70px"><i><b>a. Prerequisites:</i></b></p>

<p align="justify"; style="padding-left:70px"><i>1. Windows 10 or later: Ensure 
that you are running Windows 10 or a later version.</i></p>

<p align="justify"; style="padding-left:70px"><i>2. Enable WSL: Open PowerShell 
as Administrator and run the following command to enable WSL:<br> <code><b>dism.exe 
/online /enable-feature /featurename:Microsoft-Windows-Subsystem-Linux /all 
/norestart</code></b></i></p>

<p align="justify"; style="padding-left:70px"><i>3. Enable Virtualization: 
If you're using a virtual machine, ensure that virtualization is enabled in 
your BIOS/UEFI settings. You can check if virtualisation is enabled on your 
task manager as shown in the image below</i></p>

<p align="center"><img width="400" height="300" alt="logo"
src="assets/images/logo/comp_5.png"></p>

<p align="justify"; style="padding-left:70px"><i><b>b. Install Ubuntu 
using CLI:</i></b></p>

<p align="justify"; style="padding-left:70px"><i>1. Open PowerShell: Open 
PowerShell as Administrator.</i></p>

<p align="justify"; style="padding-left:70px"><i>2. Download Ubuntu Image: 
Download the Ubuntu image (choose the version you want) using the curl 
command: <br><code><b>curl.exe -o ubuntu-20.04.appx https://aka.ms/wslubuntu2004
</code></b> <br> Replace 
20.04 with the desired version. </i></p>

<p align="justify"; style="padding-left:70px"><i>3. Install Ubuntu: 
Install Ubuntu using the Add-AppxPackage PowerShell cmdlet: 
<br><code><b>Add-AppxPackage 
.\ubuntu-20.04.appx</code></b></i></p>

<p align="justify"; style="padding-left:70px"><i>4. Launch Ubuntu: 
After the installation is complete, launch Ubuntu by 
running: <br><code><b>ubuntu2004</code></b></i></p>

<p align="justify"; style="padding-left:70px"><i>5. Set Up Ubuntu: 
The first time you launch Ubuntu, it will ask 
you to set up a new user account with a 
username and password.</i></p>

<p align="justify"; style="padding-left:70px"><i><b>c. Update and 
Upgrade.</b> After setting up Ubuntu, it's a good idea to update the package 
lists and upgrade the installed packages to ensure you have the latest 
software versions. Open the Ubuntu terminal and run the following 
commands: <br>
<code><b>sudo apt update</code></b>
<br>
<code><b>sudo apt upgrade</code></b>
</i></b></p>


<h4>(iii) Bioinformatics tools</h4>

<p align="justify">Before we list the bioinformatics tools required to analyse genomic data, we 
will first need to understand various software management applications that we would need to run 
these tools :</p>

<h4>Software management applications</h4>

<p align="justify">Software management applications are indispensable tools in bioinformatics 
because they simplify software management, enhance reproducibility, support collaboration, and 
provide the flexibility needed to tackle the complex and dynamic nature of biological data 
analysis. These tools contribute to the efficiency and rigor of bioinformatics research and 
analysis workflows. Here, we will provide an overview of two software management 
applications, 
i.e., Docker and Conda.</p>


<h5>a. Docker</h5>

<p align="justify"; style="padding-left: 40px"><b>Overview</b></p>

<p align="justify"; style="padding-left: 40px">Docker is an open platform for developing, 
shipping, and running applications. It provides the ability to package and run an application in 
a loosely isolated environment called a container. Containers are lightweight and contain 
everything needed to run the application, so you do not need to rely on what is currently 
installed on the host; i.e., it involves bundling an application together with all of the 
necessary configuration files, libraries, and dependencies to ensure the software can run in a 
reproducible fashion across a diversity of computing environments. You can easily share 
containers while you work, and be sure that everyone you share with gets the same container that 
works in the same way.</p>

<p align="justify"; style="padding-left: 40px"><b>Docker architecture</b></p>

<p align="justify"; style="padding-left: 40px">Installing and running docker is dependent on the 
computer’s operating system. We will define some key concepts that you will come across when 
installing and running docker.</p>

<p align="center"><img width="600" height="340" alt="logo"
src="assets/images/logo/comp_3.png"></p>

<ul align="justify"; style="padding-left: 60px">
<li><b>Docker Desktop:</b> Docker Desktop is a user-friendly application for desktop and laptop 
computers that allows developers to create, manage, and run Docker containers on their local 
operating system. It simplifies the process of containerisation by providing a graphical 
interface and tools to build, test, and deploy applications within lightweight, isolated 
containers. Docker Desktop is commonly used for software development, testing, and application 
deployment, making it easier to work with containerised environments on personal computers.</li>

<li><b>Docker Engine:</b> Docker Engine is the core component of the Docker platform. It is 
responsible for creating and managing Docker containers, which are lightweight, portable, and 
self-contained environments for running applications.</li>

<li><b>Docker Daemon:</b> The Docker Daemon, often simply referred to as "the daemon," is a 
crucial component of the Docker platform. Docker daemon is the brain behind the whole operation - 
it plays a central role in managing Docker containers and orchestrating containerised 
applications.</li>

<li><b>Docker Client:</b> The Docker Client, often referred to as just "Docker," is a 
command-line tool or graphical user interface (GUI) that allows users to interact with the Docker Daemon and manage 
Docker containers, images, networks, volumes, and other Docker-related resources. It pipes/sends 
commands from your operating system terminal to the Docker environment. Docker client talks to 
Docker daemon.</li>
</ul>

<p align="justify"; style="padding-left: 40px">To install Docker on your system, you have two 
primary options: Docker Engine and Docker Desktop.</p>

<ul align="justify"; style="padding-left: 60px">

<li>If you prefer a lightweight Docker installation without additional graphical 
interfaces, Docker Engine is the choice to go with. Docker Engine can be installed on various 
Linux distributions.</li>

<li>Docker Desktop is a more user-friendly option that includes Docker Engine along with 
additional features and a graphical user interface. It's available for both Windows and 
macOS.</li>

</ul>

<p align="justify"; style="padding-left: 60px"><b><i>Note</p></b></i>

<p align="justify"; style="padding-left: 60px"><b><i>HyperKit</b> is a lightweight hypervisor 
designed for macOS operating 
systems. It is an 
open-source virtualisation tool that provides virtualisation capabilities on macOS, making it 
possible to run virtual machines (VMs) and containers on Mac computers. HyperKit is often 
associated with Docker for Mac, as it is one of the components used by Docker to enable 
containerisation and virtualisation on macOS.</I></p>

<p align="justify"; style="padding-left: 60px"><b><I>Hyper-V</b> is a virtualisation platform and 
hypervisor technology 
developed 
by Microsoft. It 
allows you to create and manage virtual machines (VMs) on Windows-based systems. Hyper-V is 
commonly used for server virtualisation, testing and development environments, and running 
multiple operating systems on a single physical machine.
</ul></i></p>

<p align="center"><img width="600" height="340" alt="logo"
src="assets/images/logo/comp_4.png"></p>


<p align="justify"; style="padding-left: 40px"><i><b>How to install 
Docker</b></i></p>

<p align="justify"; style="padding-left:60px"><i><b>a. Steps to install Docker on 
macOS:</i></b></p>

<p align="justify"; style="padding-left:80px"><i><b>1. Download Docker 
Desktop:</i></b></p>

<ul align="justify"; style="padding-left:100px">
<li><i>Go to the <a 
href="https://docs.docker.com/desktop/install/mac-install/">Docker Desktop for Mac 
page</a> on the Docker website.</i></li>

<li><i>Click the "Download from Docker 
Hub" button to download the Docker Desktop installer.</i></li>

</ul>

<p align="justify"; style="padding-left:80px"><i><b>2. Install Docker 
Desktop:</i></b></p>

<ul align="justify"; style="padding-left:100px">

<li><i>Open the downloaded <b>.dmg</b> 
file to mount the Docker Desktop disk image.</i></li>

<li><i>Drag the Docker icon to the 
Applications folder.</i></li>

</ul>

<p align="justify"; style="padding-left:80px"><i><b>3. Run Docker 
Desktop:</i></b></p>

<ul align="justify"; style="padding-left:100px">

<li><i>Open Docker Desktop from your 
Applications folder.</i></li>

</ul>

<p align="justify"; style="padding-left:80px"><i><b>4. Login to Docker Hub 
(Optional):</i></b></p>

<ul align="justify"; style="padding-left:100px">

<li><i>If you have a Docker Hub account, 
you can log in to Docker Desktop. This step is optional but allows you to access and 
download images from Docker Hub.</i></li>

</ul>

<p align="justify"; style="padding-left:80px"><i><b>5. Configure Resources 
(Optional):</i></b></p>

<ul align="justify"; style="padding-left:100px">

<li><i>In the Docker Desktop preferences, 
you can configure resources such as CPU, memory, and disk space allocated to Docker 
containers. Adjust these settings based on your system resources.</i></li>

</ul>

<p align="justify"; style="padding-left:80px"><i><b>6. Start Docker 
Desktop:</i></b></p>

<ul align="justify"; style="padding-left:100px">

<li><i>Click the Docker Desktop icon in 
your Applications folder to start Docker. The Docker icon in the menu bar indicates 
that Docker is running.</i></li>

</ul>

<p align="justify"; style="padding-left:80px"><i><b>7. Verify 
Installation:</i></b></p>

<ul align="justify"; style="padding-left:100px">

<li><i>To verify that Docker is installed 
and running, open a terminal and run the following commands:</i></li>

<p align="justify"; style="padding-left:80px"><code><b>docker 
--version</code></b></p>

<p align="justify"; style="padding-left:80px"><code><b>docker run 
hello-world</code></b></p>

<p align="center"><img width="400" height=“400” alt="logo"
src="assets/images/logo/comp_6.png"></p>

<br>
<li><i>The first command should display 
the Docker version, and the second command runs a simple container to verify that 
Docker can pull and run images.</i></li>

</ul>

<p align="justify"; style="padding-left:60px"><i><b>b. Steps to install Docker on 
ubuntu:</I></b></p>

<p align="justify"; style="padding-left:60px">Run the following commands on you 
ubuntu terminal</p>

<p align="justify"; style="padding-left:80px"><i><b>1. Update Package 
Lists:</i></b></p>

<p align="justify"; style="padding-left:80px"><code><b>sudo apt 
update</code></b></p>

<p align="justify"; style="padding-left:80px"><i><b>2. Install 
Prerequisites:</i></b></p>

<p align="justify"; style="padding-left:80px"><code><b>sudo apt install 
apt-transport-https ca-certificates curl software-properties-common</code></b></p>

<p align="justify"; style="padding-left:80px"><i><b>3. Add Docker's GPG 
Key:</i></b></p>

<p align="justify"; style="padding-left:80px"><code><b>curl -fsSL 
https://download.docker.com/linux/ubuntu/gpg | sudo gpg --dearmor -o 
/usr/share/keyrings/docker-archive-keyring.gpg</code></b></p>

<p align="justify"; style="padding-left:80px"><i><b>4. Set Up the Stable Docker 
Repository:</i></b></p>

<p align="justify"; style="padding-left:80px"><i>For Ubuntu 20.04 (Focal 
Fossa):</i></p>

<p align="justify"; style="padding-left:80px"><code><b>echo "deb [arch=amd64 
signed-by=/usr/share/keyrings/docker-archive-keyring.gpg] 
https://download.docker.com/linux/ubuntu focal stable" | sudo tee 
/etc/apt/sources.list.d/docker.list > /dev/null</code></b></p>

<p align="justify"; style="padding-left:80px"><i><b>5. Install Docker 
Engine:</i></b></p>

<p align="justify"; style="padding-left:80px"><code><b>sudo apt 
update</code></b></p>

<p align="justify"; style="padding-left:80px"><code><b>sudo apt install docker-ce 
docker-ce-cli containerd.io</code></b></p>

<p align="justify"; style="padding-left:80px"><i><b>6. Start and Enable 
Docker:</i></b></p>

<p align="justify"; style="padding-left:80px"><code><b>sudo systemctl start 
docker</code></b></p>

<p align="justify"; style="padding-left:80px"><code><b>sudo systemctl enable 
docker</code></b></p>

<p align="justify"; style="padding-left:80px"><i><b>7. Verify Docker 
Installation:</i></b></p>

<p align="justify"; style="padding-left:80px"><code><b>docker 
--version</code></b></p>

<p align="justify"; style="padding-left:80px"><i><b>8. Run a Test 
Container:</i></b></p>

<p align="justify"; style="padding-left:80px"><code><b>docker run 
hello-world</code></b></p>

<p align="justify"; style="padding-left:80px"><i>If everything is set up correctly, 
you should see a message indicating that your Docker installation is 
working.</i></p>

<p align="justify"; style="padding-left:80px"><i><b>9. Manage Docker as a Non-root 
User (Optional):</i></b></p>

<p align="justify"; style="padding-left:80px"><i>To avoid using <b>sudo</b> with 
every Docker command, add your user to the <b>docker</b> group:</i></p>

<p align="justify"; style="padding-left:80px"><code><b>sudo usermod -aG docker 
$USER</code></b></p>

<p align="justify"; style="padding-left:80px"><i>Log out and log back in or restart 
your system for the changes to take effect.</i></p>

<br>
<p align="justify"; style="padding-left: 40px"><i><b>Docker Usage</b></i></p>  

<p align="justify"; style="padding-left:60px"><i>In this course, we will be using 
Docker as a our software management application and Docker images (bioinformatics 
tools) built and catalogued by Sanger Pathogens, the State Public health 
Bioinformatics (StaPH-B) and NCBI.</i></p>

<p align="justify"; style="padding-left:80px"><i><b>Step 1: Downloading a Docker 
image</i></b></p>

<p align="justify"; style="padding-left:80px"><i>Download an image from a repository  
by using the <code><b>docker pull</b></code> command: </i></p>

<p align="justify"; style="padding-left:80px"><code><b>docker pull 
repository/(name_of_container):(tag)</b></code></p>

<p align="justify"; style="padding-left:80px"><i>For example</i></p>

<p align="justify"; style="padding-left:80px"><code><b>docker pull 
staphb/snippy</b></code></p>

<p align="justify"; style="font-size: 12px; padding-left:80px"><b><i>An explanation 
of 
this command is as
follows:</i></p></b>

<p align="justify"; style="font-size: 12px; padding-left:80px"><i><b>docker 
pull:</b> instructs Docker to download an image from a repository</I></p>

<p align="justify"; style="font-size: 12px; 
padding-left:80px"><i><b>staphb/snippy:</b> is the docker image</I></p>

<p align="justify"; style="padding-left:80px"><I>When you pull a docker image, you 
should see an output similar to this…</I></p>

<p align="center"><img width="400" height="180" alt="logo"
src="assets/images/logo/comp_7.png"></p>

<p align="justify"; style="padding-left:80px"><i><b>Step 2: Running a Docker
image</i></b></p>

<p align="justify"; style="padding-left:80px"><i>Starting a container from a docker 
image is simply done using the command <code><b>docker run</code></b>. A brand new 
Docker container is then created, and will run any command you provide within the 
container.</i></p>

<p align="justify"; style="padding-left:80px"><i>For example</i></p>

<p align="justify"; style="padding-left:80px"><code><b>docker run --rm=True -v 
$PWD:/data 
-u $(id -u):$(id -g) staphb/fastqc fastqc 21999_7#176*.fastq.gz 
</b></code></p>

<p align="justify"; style="font-size: 12px; padding-left:80px"><b><i>An explanation
of
this command is as
follows:</i></p></b>

<p align="justify"; style="font-size: 12px; padding-left:80px"><i><b>docker run:</b> 
function to start a new container </I></p>


<p align="justify"; style="font-size: 12px; padding-left:80px"><i><b>--rm=True:</b> 
By default, when a Docker container is run without this flag, the Docker container 
is created,the container runs, and then exits, but is not deleted. In other words, 
Docker containers are NOT ephemeral by default. A local copy of the container is 
kept and takes up unnecessary storage space. It is a good idea to always use this 
flag so that the container is removed after running it, unless for some reason you 
need the container after the specified program has been run.</I></p>

<p align="justify"; style="font-size: 12px; padding-left:80px"><i><b>-v 
$PWD:/data:</b> The <b>-v</b> flag mounts a volume between your local machine and 
the Docker container. This specific command mounts the present working directory to 
the /data directory within the Docker container, which makes the files on your local 
machine accessible to the container. You can change these paths to meet the needs of 
your system, however it is a good idea to have a working directory in each of the 
containers, and thus each container contains the /data directory for such 
purpose.</I></p>

<p align="justify"; style="font-size: 12px; padding-left:80px"><i><b>--u $(id 
-u):$(id -g):</b> By default, when Docker containers are run, they are run as the 
root user. This can be problematic because any files created from within the 
container will have root permissions/ownership and the local user will not be able 
to do much with them. The -u flag sets the container's user and group based on the 
user and group from the local machine, resulting in the correct file 
ownership.</I></p>


<p align="justify"; style="font-size: 12px;
padding-left:80px"><i><b>staphb/fastqc:</b> is the docker image. [ <b>staphb</b> - 
is the repository; <b>fastqc</b> - is the name of the container image]</I></p>

<p align="justify"; style="font-size: 12px;
padding-left:80px"><i><b>fastqc :</b> Command to run in that container</I></p>

<p align="justify"; style="font-size: 12px;
padding-left:80px"><i><b>21999_7#176*.fastq.gz:</b> Option(s) of the command (paths 
in options are based on container file system).</I></p>

<p align="justify"; style="padding-left:80px"><i>To avoid long commands, the [docker 
run --rm=True -v $PWD:/data -u $(id -u):$(id -g)] command can be incorporated into a 
bash function by including the following into your ~/.bashrc (if you are using Bash 
shell / Linux) or ~/.zshrc (if you are using Z shell / macOS)</i></p>

<p align="justify"; style="padding-left:80px"><i><b># For Bash Shell,</b> 
run the command:</i></p>

<p align="justify"; style="padding-left:80px"><i><b><code>echo 'function 
docker_run() { docker run --rm=True -u $(id -u):$(id -g) -v $(pwd):/data "$@" ;}' >> 
~/.bash_profile</code></b></i></p>

<p align="justify"; style="padding-left:80px"><i><b># For Z Shell,</b> run 
the command:</i></p>

<p align="justify"; style="padding-left:80px"><i><b><code>echo 'function 
docker_run() { docker run --rm=True -u $(id -u):$(id -g) -v $(pwd):/data "$@" ;}' >> 
~/.zshrc</b></i></code></p>

<p align="justify"; style="padding-left:80px"><i>You need to be in your home 
directory and then run the one of the following commands depending on your 
shell</i></p>

<p align="justify"; style="padding-left:80px"><i>You need to restart the Terminal 
for this change to take effect or Source the Bashrc by running the command:</i></p>

<p align="justify"; style="font-size: 12px;
padding-left:80px"><i><b><code>source ~/.bash_profile</b></code></i></p>

<p align="justify"; style="padding-left:80px"><i>To check the above command has 
worked properly open bashrc file using:
<code><b>cat .bash_profile</code></b> or <code><b>cat .zshrc</code></b></i></p>

<br>
<p align="justify"; style="padding-left: 40px"><i><b>Docker 
images/Bioinformatics tools</b></i></p>

<p align="justify"; style="padding-left:60px"><i>We will use various 
bioinformatics tools in this course, and the table below shows a list of 
tools required for this course. These tools can be accessed and executed 
using Docker platform.</i></p>

<br>

<style>
table, th, td {
  border: 1px solid black;
  border-collapse: collapse;
}
</style>

<table style="font-size: 12px; width:100%">
  <tr>
    <th>Tool</th>
    <th>Website</th> 
    <th>Brief Description</th>
    <th>Installation</th>
  </tr>
  <tr>
    <td>FastQC</td>
    <td>https://github.com/s-andrews/FastQC</td>
    <td>Quality assessment for high throughput sequencing datasets</td>
    <td>docker pull staphb/fastqc</td>
  </tr>
<tr>
    <td>Trimmomatic</td> 
    <td>https://github.com/usadellab/Trimmomatic</td>
    <td>Adapter trimming</td>
    <td>docker pull staphb/trimmomatic</td>
  </tr>
<tr>
    <td>bwa</td>
    <td>https://github.com/lh3/bwa</td>
    <td>Mapping DNA sequences</td>
    <td>docker pull staphb/bwa</td>
  </tr>
<tr>
    <td>samtools</td>
    <td>https://github.com/samtools/</td>
    <td>Manipulating NGS</td>
    <td>docker pull staphb/samtools</td>
  </tr>
<tr>
    <td>SPAdes</td>
    <td>https://github.com/ablab/spades/releases</td>
    <td><i>De novo</i> assembly</td>
    <td>docker pull staphb/spades</td>
  </tr>
<tr>
    <td>snp-sites</td>
    <td>https://github.com/sanger-pathogens/snp-sites</td>
    <td>Rapidly extract SNPs from a multi-FASTA alignment</td>
    <td>docker pull staphb/snp-sites</td>
  </tr>
<tr>
    <td>QUAST</td>
    <td>https://github.com/ablab/quast</td>
    <td>Quality assessment for genome assemblies</td>
    <td>docker pull staphb/quast</td>
  </tr>
<tr>
    <td>BLAST</td>
    <td>https://github.com/ncbi/blast_plus_docs</td>
    <td>Compares nucleotide or protein sequences to sequence databases and 
calculates the statistical significance of matches</td>
    <td>docker pull ncbi/blast</td>
  </tr>
<tr>
    <td>Kraken2</td>
    <td>https://github.com/DerrickWood/kraken2</td>
    <td>Kraken is an ultrafast and highly accurate program for assigning taxonomic 
labels to metagenomic DNA sequences</td>
    <td>docker pull staphb/kraken2</td>
  </tr>
<tr>
    <td>snippy</td>
    <td>https://github.com/tseemann/snippy</td>
    <td>Rapid haploid variant calling and core genome alignment</td>
    <td>docker pull staphb/snippy</td>
  </tr>
<tr>
    <td>FastTree</td>
    <td>http://www.microbesonline.org/fasttree/</td>
    <td>Phylogenetic tree builder</td>
    <td>docker pull staphb/fasttree</td>
  </tr>
<tr>
    <td>Gubbins</td>
    <td>https://github.com/nickjcroucher/gubbins</td>
    <td>Builds phylogeny after removing regions of recombination</td>
    <td>docker pull sangerpathogens/gubbins</td>
  </tr>
<tr>
    <td>popPUNK</td>
    <td>https://github.com/bacpop/PopPUNK</td>
    <td>Clustering</td>
    <td>docker pull staphb/poppunk</td>
  </tr>
<tr>
    <td>Abricate</td>
    <td>https://github.com/tseemann/abricate</td>
    <td>Mass screening of contigs for antimicrobial resistance or virulence 
genes</td>
    <td>docker pull staphb/abricate</td>
  </tr>
<tr>
    <td>Prokka</td>
    <td>https://github.com/tseemann/prokka</td>
    <td>Prokka is a software tool to annotate bacterial, archaeal and viral genomes 
quickly and produce standards-compliant output files</td>
    <td>docker pull staphb/prokka</td>
  </tr>
<tr>
    <td>Any2fasta</td>
    <td>https://github.com/tseemann/any2fasta</td>
    <td>Convert various sequence formats to FASTA</td>
    <td>docker pull staphb/any2fasta</td>
  </tr>
<tr>
    <td>ARIBA</td>
    <td>https://github.com/sanger-pathogens/ariba</td>
    <td>Antimicrobial Resistance Identification By Assembly</td>
    <td>docker pull staphb/ariba</td>
  </tr>
<tr>
    <td>SeroBA</td>
    <td>https://github.com/sanger-pathogens/seroba</td>
    <td>Identify the Serotype from Illumina NGS reads for given references</td>
    <td>docker pull staphb/seroba</td>
  </tr>
<tr>
    <td>SRST2</td>
    <td>https://github.com/katholt/srst2</td>
    <td>Gene detection and MLST</td>
    <td>docker pull staphb/srst2</td>
  </tr>
<tr>
    <td>mlst</td>
    <td>https://github.com/tseemann/mlst</td>
    <td>Scan contig files against traditional PubMLST typing schemes</td>
    <td>docker pull staphb/mlst</td>
  </tr>  
</table>

<br>
<p align="justify"; style="font-size: 12px; padding-left:80px"><i>To check that 
Docker images have been downloaded, run the command:</I></p>

<p align="justify"; style="padding-left:80px"><i><b><code>docker images | grep 
"staphb\|sanger\|ncbi"| wc -l</code></b></i></p>

<p align="justify"; style="font-size: 12px; padding-left:80px"><i>There should be 
20</i></p>

<br>
 
<h5>b. Conda</h5>

<p align="justify"; style="padding-left: 40px"><b>Definitions</b></p>

<p align="justify"; style="padding-left: 40px"><b>Anaconda:</b>is a distribution of 
Python and R programming languages for scientific computing, data science, machine 
learning, and related domains. It includes a variety of pre-installed libraries and 
tools commonly used in these fields. Anaconda simplifies the process of managing 
packages and environments for data science projects. Simply, Anaconda is a 
comprehensive distribution that includes a variety of pre-installed packages for 
data science and scientific computing.</p>

<p align="justify"; style="padding-left: 40px"><b>Miniconda:</b>is a minimal 
installer for the Conda package manager. Unlike Anaconda, which comes with a 
pre-selected set of packages, Miniconda allows users to install only the packages 
they need, making it a lightweight alternative. Simply, Miniconda is a minimal 
installer for Conda, allowing users to install only the packages they need for a 
more lightweight setup.</p>

<p align="justify"; style="padding-left: 40px"><b>Conda:</b> is the package manager 
used by both Anaconda and Miniconda. It is an open-source package management and 
environment management system that runs on Windows, macOS, and Linux. Simply, Conda 
is the package manager that can be used independently of Anaconda or Miniconda. It 
simplifies the process of managing packages and environments.</p>

<p align="justify"; style="padding-left: 40px"><b>Conda environment:</b> A Conda 
environment is a self-contained directory that contains a specific collection of 
Python (or other programming language) packages and their dependencies. Environments 
allow you to manage and isolate different sets of packages, ensuring that your 
projects have the specific dependencies they need without interfering with each 
other.</p>

<p align="justify"; style="padding-left:60px"><i><b>i. Steps to install 
Anaconda:</I></b></p>

<p align="justify"; style="padding-left:80px"><i><b>1. Download:</i></b></p>

<ul align="justify"; style="padding-left:100px">
<li><i>Visit the <a
href="https://www.anaconda.com/download">Anaconda download page</a> and choose the 
appropriate version (Python 3.x is recommended).</i></li>

</ul>

<p align="justify"; style="padding-left:80px"><i><b>2. Install:</i></b></p>

<ul align="justify"; style="padding-left:100px">
<li><i>Follow the installation instructions for your operating system. The 
installation typically involves running an installer and accepting the license 
terms.</i></li>

</ul>

<p align="justify"; style="padding-left:80px"><i><b>3. Verify 
Installation:</i></b></p>

<ul align="justify"; style="padding-left:100px">

<li><i>After installation, you can open a terminal or command prompt and 
run:</i></li>

<p align="justify"; style="padding-left:80px"><code><b>conda 
--version</code></b></p>

<li><i>This command should display the installed version of conda.</i></li>

</ul>

<p align="justify"; style="padding-left:80px"><i><b>4. Create and Activate an 
Environment (Optional):</i></b></p>

<ul align="justify"; style="padding-left:100px">

<li><i>You can create a new environment using:</i></li>

<p align="justify"; style="padding-left:80px"><code><b>conda create --name 
myenv</code></b></p>

<i>Replace "myenv" with the desired name for your environment.</i>

<li><i>Activate the environment:</i></li>

<p align="justify"; style="padding-left:80px"><code><b>conda activate 
myenv</code></b></p>

</ul>

<p align="justify"; style="padding-left:60px"><i><b>ii. Steps to install 
Miniconda:</I></b></p>

<p align="justify"; style="padding-left:80px"><i><b>1. Download and 
Install:</I></b></p>

<ul align="justify"; style="padding-left:100px">
<li><i>Visit the <a
href="https://docs.conda.io/projects/miniconda/en/latest/">Miniconda website 
page</a> and then follow the installation instructions.</i></li>

</ul>

<p align="justify"; style="padding-left:80px"><i><b>2. Create and Activate an 
Environment (Optional):</i></b></p>

<ul align="justify"; style="padding-left:100px">

<li><i>You can create a new environment using:</i></li>

<p align="justify"; style="padding-left:80px"><code><b>conda create --name 
myenv</code></b></p>

<i>Replace "myenv" with the desired name for your environment.</i>

<li><i>Activate the environment:</i></li>

<p align="justify"; style="padding-left:80px"><code><b>conda activate 
myenv</code></b></p>

</ul>

<p align="justify"; style="padding-left:80px"><i><b>3. Install Packages:</i></b></p>

<ul align="justify"; style="padding-left:100px">

<li><i>Once the environment is activated, you can use conda to install packages. For 
example:</i></li>

<p align="justify"; style="padding-left:80px"><code><b>conda install 
numpy</code></b></p>

</ul>

<p align="justify"; style="padding-left:80px"><i><b>3. Deactivate the 
Environment:</i></b></p>

<ul align="justify"; style="padding-left:100px">

<li><i>When you're done working in the environment, you can deactivate it:</i></li>

<p align="justify"; style="padding-left:80px"><code><b>conda 
deactivate</code></b></p>

</ul>

<p align="justify"; style="padding-left:80px"><i><b>4. Remove an Environment 
(Optional):</i></b></p>

<ul align="justify"; style="padding-left:100px">

<li><i>If you want to remove an environment, you can use:</i></li>

<p align="justify"; style="padding-left:80px"><code><b>conda env remove --name 
myenv</code></b></p>

</ul>

<p align="justify"; style="padding-left:80px"><i><b>N/B:</B>Miniconda is a great 
choice for users who prefer a more minimalistic and customizable setup. It allows 
you to build environments tailored to specific projects or use cases without the 
overhead of a larger distribution like Anaconda.</I></p>

 <script>
/* Loop through all dropdown buttons to toggle between hiding and showing its dropdown content - This allows the user to 
have multiple dropdowns without any conflict */
var dropdown = document.getElementsByClassName("dropdown-btn");
var i;

for (i = 0; i < dropdown.length; i++) {
  dropdown[i].addEventListener("click", function() {
    this.classList.toggle("active");
    var dropdownContent = this.nextElementSibling;
    if (dropdownContent.style.display === "block") {
      dropdownContent.style.display = "none";
    } else {
      dropdownContent.style.display = "block";
    }
  });
}
</script>
</body>
</html> 



