---
layout: home
# articles:
#   excerpt_type: html
---
<!DOCTYPE html>
<html>
<head>
<meta name="viewport" content="width=device-width, initial-scale=1">
<style>
body {
  font-family: "Lato", sans-serif;
}

.sidenav {
  height: 90%;
  width: 400px;
  position: fixed;
  z-index: 1;
  top: 40;
  right: 0;
  background-color: #0C6980;
  overflow-x: hidden;
  padding-top: 40px;
  transition: 0.5s;
}

.sidenav a, .dropdown-btn {
  padding: 10px 20px 10px 10px;
  text-decoration: none;
  font-size: 12px;
  color: white;
  display: block;
  cursor: pointer; 
  outline: none;
  border: none;
  background: none;  
  width: 100%;  
  text-align: right;
  transition: 0.3s;
 

}

.sidenav a:hover, .dropdown-btn:hover {
  color: #1e504f;
}

.sidenav .closebtn {
  position: absolute;
  top: 0;
  right: 0px;
  font-size: 20px;
  right: 0px;
}

.main {
  margin-right: 450px; /* Same as the width of the sidenav */
  font-size: 14px; /* Increased text to enable scrolling */
  padding: 0px 10px;
}

.active {  
  background-color: #77A7B0;
  color: white;
} 

.dropdown-container {
  display: none;
  background-color: #6CB0A8;
  padding-left: 8px;
}

.fa-caret-down {
  float: left;
  padding-left: 8px;
}


@media screen and (max-height: 450px) {
  .sidenav {padding-top: 15px;}
  .sidenav a {font-size: 18px;}
}
</style>
</head>
<body>

<br>
<b><span style="font-size:14px;cursor:pointer;font-family:sans-serif" onclick="openNav()">&#9776; Content</span></b>


<div id="mySidenav" class="sidenav">
  <a href="javascript:void(0)" class="closebtn" onclick="closeNav()">&times;</a>
  <a href="Bioinformatics_Workshop.html">Course Overview</a>

  <button class="dropdown-btn">Data, Tools and Computational Platforms
    <i class="fa fa-caret-down"></i>
  </button>
  <div class="dropdown-container">
    <a href="computing.html">Overview</a>
    <a href="comp_specs.html">Computer Specifications</a>
    <a href="docker.html">Software Management Apps (Docker) & Bioinformatics Tools</a>
  </div>
  
<button class="dropdown-btn">Genomic Data Analysis (Bioinformatics)
    <i class="fa fa-caret-down"></i>
  </button>
  <div class="dropdown-container">
    <a href="analysis.html">Overview</a>
    <a href="ngs_file_formats.html">NGS File Formats</a>
    <a href="ngs_data_QC.html">NGS Data Quality Control</a>
    <a href="adapter_trimming.html">Adapter Trimming</a>
    <a href="taxonomy.html">Taxonomy Sequence Classification</a>
    <a href="denovo_assembly.html">de novo Assembly</a>
    <a href="quality_assessment_genome_assemblies.html">Quality Assessment
of Genome Assemblies</a>
    <a href="genome_annotation.html">Genome Annotation</a>
    <a href="serotyping.html">Serotyping</a>
    <a href="mlst.html">Multilocus Sequence Typing</a>
    <a href="amr.html">AMR Profiling</a>
    <a href="mapping.html">Mapping</a>
    <a href="variant_calling.html">Variant Calling</a>
    <a href="phylogenetics.html">Phylogenetics</a>
  </div>   
  
<a href="trainers.html">Acknowledgement</a>
  
</div>

<script>
  function openNav() {
    document.getElementById("mySidenav").style.width = "400px";
  }
  
  function closeNav() {
    document.getElementById("mySidenav").style.width = "0";
  }
  
  /* Loop through all dropdown buttons to toggle between hiding and showing its dropdown content - This allows the user to have multiple dropdowns without any conflict */
  var dropdown = document.getElementsByClassName("dropdown-btn");
  var i;
      
  for (i = 0; i < dropdown.length; i++) {
    dropdown[i].addEventListener("click", function() {  
      this.classList.toggle("active");
      var dropdownContent = this.nextElementSibling;
      if (dropdownContent.style.display === "block") {  
  
        dropdownContent.style.display = "none";
      } else {
        dropdownContent.style.display = "block";
      }
    });
  } 
  
  </script>

<div class="main">   
<br>
<br>

<p align=“left”><I>Before you being this section, navigate to the <a 
href="https://drive.google.com/drive/folders/1X_Mu0B94Z5Xf1jP6qfRHNNbGwJthzXHh?usp=share_link">denovo_assembly</a> folder. You will use this folder 
and
its contents to learn and
practice this section.</p></I>

<br>

<h4>Overview</h4>
<p align="justify">After generating sequencing reads and carrying out quality 
assessment, the next step is to determine how the reads fit together by 
looking for overlaps between them; this process is called <b>genome 
assembly</b>.</p>

<p align="justify">The data from the Illumina machine comes as relatively 
short stretches (35 - 150 base pairs) of DNA – around 6 billion of them. 
These individual sequences are called sequencing reads. There are a range 
of assembly programs that have been specifically designed to assemble 
genomes from sequence reads data. Genome assembly using sequence reads of 
around 100bp is complicated due to the high frequency of repeats longer 
than the sequence read length in genomes, for example: IS elements, rRNA 
operons; and the massive amount of data the programs have to handle. In 
addition to finding overlaps in the sequence, the assembly programs can 
also use information from the predicted insert size where paired reads are 
used, to link and position reads in an assembly.</p>

<p align="justify">Where a genome is pieced together without any reference 
sequence to compare it to, or scaffold it against, it is termed a de novo 
assembly. De novo assembly may not produce complete genomes but will be 
fragmented into multiple contiguous sequences (contigs), the order of 
which is arbitrary, and does not necessarily bear not any relation to 
their real order in the genome.</p>

<p align="justify">Where a closely related reference sequence is available, 
it is possible to improve an assembly by ordering the contigs in 
comparison to the reference, and also transferring annotation. In this 
case, nearly all of the genome will be present, i.e. genes and features, 
but there will be some regions that will contain gaps, or contigs that 
will not be accurately placed, because they are not present in the 
reference used. Although technically incomplete, ordered genome assemblies 
can provide valuable insights into the genetics and biology of an 
organism.</p>
 
<h4>Tool(s)</h4>
<p align="justify">In this section we will use SPAdes which is one of a 
number of<i> de novo</i> assemblers that use short read sets as input (e.g. 
Illumina Reads), and the assembly method is based on de Bruijn graphs.</p>

<p align="justify">You 
can download SPAdes tool from a docker repository using the command:</p>
 
<br>

<code><b>docker pull staphb/spades</code></b>
 
 
<h4>Generating a de novo assembly for a single strain</h4>

<p align="justify">We will assemble the genome of Streptococcus pneumoniae 
strain <b>17150_4#79</b> (accession number: ERR1065718). Run the command in terminal to execute SPAdes:</p>

<br>

<code><b>docker_run staphb/spades spades.py -1 17150_4#79_1.fastq.gz -2 
17150_4#79_2.fastq.gz --careful --cov-cutoff auto -o 
spades_assembly</code></b>
  
 
<p style="font-size: 12px; padding-left: 30px"><b>An explanation of this 
command is as
follows:</b></p>
<p style="font-size: 12px; padding-left: 30px"><b>docker_run:</b> is a 
function to start a container. The function includes the
following flags: docker run --rm=True -u $(id -u):$(id -g) -v $(pwd):/data 
"$@". To understand
the docker_run function read the section [Data and Computational 
Platforms (Docker)]</p>
<p style="font-size: 12px; padding-left: 30px"><b> staphb/spades:</b> is 
the docker image</p>
<p style="font-size: 12px; padding-left: 30px"><b> spades.py:</b> is the 
tool</p>
<p style="font-size: 12px; padding-left: 30px"><b>-1:</b> flag for the 
input file of forward reads </p>
<p style="font-size: 12px; padding-left: 30px"><b>-2:</b> flag for the 
input file of reverse reads</p>
<p style="font-size: 12px; padding-left: 30px"><b>--careful:</b> minimizes 
mismatches and short indels</p>
<p style="font-size: 12px; padding-left: 30px"><b>--cov-cutoff auto:</b> 
computes the coverage threshold (rather than the default setting, 
“off”)</p>
<p style="font-size: 12px; padding-left: 30px"><b>-o:</b> flag for the the 
output directory</p>

<p align="justify">Once SPAdes has finished running, you will see the output,</p>

<p align="center"><img width="400" height="240" alt="logo"
src="assets/images/logo/denovo_1.png"></p>

<p align="justify">Move into the output directory (spades_assembly) and look 
at the contigs.fa file usimg the command:</p>
<br>
<code><b>head contigs.fasta</code></b>
<br>

<p align="center"><img width="400" height="240" alt="logo"
src="assets/images/logo/denovo_2.png"></p>


<h4>Generating a <i>de novo</i> assembly for a multiple strains</h4>

<p align="justify">We can also assemble genomes of multiple strains.  
Navigate to the multiple_files folder and carry out the steps below.</p> 
<p align="justify"><b>Step 1:</b> We will first  create a folder for each pair 
of compressed fastq files and named after the strain id using the 
command:</p>
<br>
<code><b>for x in *1.fastq.gz; do mkdir ${x%%_1.fastq.gz} ; mv $x 
${x%%_1.fastq.gz}; mv ${x%%1.fastq.gz}2.fastq.gz ${x%%_1.fastq.gz}; 
done</code></b>

<p style="font-size: 12px; padding-left: 30px"><b>An explanation of this command is as follows:</b></p>
<p style="font-size: 12px; padding-left: 30px"><b>for x in *1.fastq.gz; do:</b> This starts a loop where <b>x</b> takes on the value of each file that 
matches the pattern "*1.fastq.gz" in the current directory.</p>
<p style="font-size: 12px; padding-left: 30px"><b> mkdir ${x%%_1.fastq.gz}:</b> This creates a directory using the prefix of the file name (i.e., 
removes "_1.fastq.gz" from the end of the file name).</p>
<p style="font-size: 12px; padding-left: 30px"><b> mv $x ${x%%_1.fastq.gz}:</b> This moves the file with "1.fastq.gz" to the directory created in the 
previous step.</p>
<p style="font-size: 12px; padding-left: 30px"><b>mv ${x%%1.fastq.gz}2.fastq.gz ${x%%_1.fastq.gz}:</b> This moves the corresponding "2.fastq.gz" file 
to the same directory.</p>

<p style="font-size: 12px; padding-left: 30px"><b>Here's a brief explanation of the <b>${x%%_1.fastq.gz} </b>syntax:</b></p>

<p style="font-size: 12px; padding-left: 30px"><b>${x}:</b> This refers to the value of the variable <b>x</b>.</p>

<p style="font-size: 12px; padding-left: 30px"><b>%%:</b> This is a pattern removal operator</p>

<p style="font-size: 12px; padding-left: 30px"><b>_1.fastq.gz:</b> This is the pattern to be removed.</p>

<p style="font-size: 12px; padding-left: 30px">So, <b>${x%%_1.fastq.gz}</b> removes the trailing "_1.fastq.gz" from the value of <b>x</b>.</p>
 
<p align=”left><b>Step 2:</b> We will then execute SPAdes using the 
command:</p>
<br>
<code><b>for x in *#* ; do docker_run staphb/spades spades.py --pe1-1 
$x/${x}_1.fastq.gz --pe1-2 $x/${x}_2.fastq.gz --careful --cov-cutoff auto 
-o $x"_output"; done</code></b>

<p style="font-size: 12px; padding-left: 30px"><b>An explanation of this command is as follows:</b></p>

<p style="font-size: 12px; padding-left: 30px"><b>for x in *#* ; do:</b> This initiates a loop, iterating over files or directories with names 
containing  <b>#</b>.</p>

<p style="font-size: 12px; padding-left: 30px"><b>docker_run staphb/spades:</b> This indicates that you are using a Docker container 
(<b>staphb/spades</b>) to run the SPAdes genome assembler.</p>

<p style="font-size: 12px; padding-left: 30px"><b>spades.py --pe1-1 $x/${x}_1.fastq.gz --pe1-2 $x/${x}_2.fastq.gz --careful --cov-cutoff auto -o 
$x"_output":</b> These are the SPAdes command-line options:
<ul style="font-size: 12px; padding-left: 50px"> 
<li><b>spades.py:</b> The SPAdes command itself.</li>
<li><b>--pe1-1 $x/${x}_1.fastq.gz:</b> The path to the first paired-end FASTQ file.</li>
<li><b>--pe1-2 $x/${x}_2.fastq.gz:</b> The path to the second paired-end FASTQ file.</li>
<li><b>-careful:</b> Activates the "careful" mode, which can be useful for low-coverage datasets.</li>
<li><b>--cov-cutoff auto:</b> Automatically sets the coverage cutoff.</li>
<li><b>-o $x"_output":</b> Specifies the output directory for the SPAdes assembly.</li>
</ul>
</p>

<p align="justify">We will use contigs generated from this section to 
complete Section Four (detecting AMR genes using ABRicate).</p> 

<p align="center"; style="color:green"><i><b>Quiz</b></i></p>
<p align="center"; style="color:green"><i><b>Navigate to the <a 
href="https://drive.google.com/drive/folders/1fRpQCX3qq-Y-UNMvWVnSwrCbCxEhhG3D?usp=share_link">s.agalactiae</a> folder. Explore the content of this 
folder and carry out:
<ul align="center"; style="color:green">
<ol><i>de novo</i> assembly for a single strain</ol>
<ol><i>de novo</i> assembly for multiple strains </ol>
</ul>
</b></i></p>
  
 </div>
    

</body>
</html>

