---
layout: home
# articles:
#   excerpt_type: html
---
<!DOCTYPE html>
<html>
<head>
<meta name="viewport" content="width=device-width, initial-scale=1">
<style>
body {
  font-family: "Lato", sans-serif;
}

.sidenav {
  height: 100%;
  width: 400px;
  position: fixed;
  z-index: 1;
  top: 0;
  right: 0;
  background-color: #0C6980;
  overflow-x: hidden;
  padding-top: 40px;
  transition: 0.5s;
}

.sidenav a, .dropdown-btn {
  padding: 10px 20px 10px 10px;
  text-decoration: none;
  font-size: 12px;
  color: white;
  display: block;
  cursor: pointer; 
  outline: none;
  border: none;
  background: none;  
  width: 100%;  
  text-align: right;
  transition: 0.3s;
 

}

.sidenav a:hover, .dropdown-btn:hover {
  color: #1e504f;
}

.sidenav .closebtn {
  position: absolute;
  top: 0;
  right: 0px;
  font-size: 20px;
  right: 0px;
}

.openbtn {
  font-size: 20px;
  cursor: pointer;
  background-color: #7cc7bb;
  color: white;
  padding: 10px 15px;
  border: none;
}

.main {
  margin-right: 450px; /* Same as the width of the sidenav */
  font-size: 14px; /* Increased text to enable scrolling */
  padding: 0px 10px;
}

.active {  
  background-color: #77A7B0;
  color: white;
} 

.dropdown-container {
  display: none;
  background-color: #7cc7bb;
  padding-left: 8px;
}

.fa-caret-down {
  float: left;
  padding-left: 8px;
}


@media screen and (max-height: 450px) {
  .sidenav {padding-top: 15px;}
  .sidenav a {font-size: 18px;}
}
</style>
</head>
<body>

<br>

<b><button class="openbtn"; style="font-size:14px;cursor:pointer;font-family:sans-serif" onclick="openNav()";>☰ Content</button></b> 


<div id="mySidenav" class="sidenav">
  <a href="javascript:void(0)" class="closebtn" onclick="closeNav()">&times;</a>
  <a href="Bioinformatics_Workshop.html">Course Overview</a>

  <button class="dropdown-btn">Data, Tools and Computational Platforms
    <i class="fa fa-caret-down"></i>
  </button>
  <div class="dropdown-container">
    <a href="computing.html">Overview</a>
    <a href="comp_specs.html">Computer Specifications</a>
    <a href="docker.html">Software Management Apps (Docker) & Bioinformatics Tools</a>
  </div>
  
<button class="dropdown-btn">Genomic Data Analysis (Bioinformatics)
    <i class="fa fa-caret-down"></i>
  </button>
  <div class="dropdown-container">
    <a href="analysis.html">Overview</a>
    <a href="ngs_file_formats.html">NGS File Formats</a>
    <a href="ngs_data_QC.html">NGS Data Quality Control</a>
    <a href="adapter_trimming.html">Adapter Trimming</a>
    <a href="taxonomy.html">Taxonomy Sequence Classification</a>
    <a href="denovo_assembly.html">de novo Assembly</a>
    <a href="quality_assessment_genome_assemblies.html">Quality Assessment
of Genome Assemblies</a>
    <a href="genome_annotation.html">Genome Annotation</a>
    <a href="serotyping.html">Serotyping</a>
    <a href="mlst.html">Multilocus Sequence Typing</a>
    <a href="amr.html">AMR Profiling</a>
    <a href="mapping.html">Mapping</a>
    <a href="variant_calling.html">Variant Calling</a>
    <a href="phylogenetics.html">Phylogenetics</a>
  </div>   
  
<a href="trainers.html">Acknowledgement</a>
  
</div>

<script>
  function openNav() {
    document.getElementById("mySidenav").style.width = "400px";
  }
  
  function closeNav() {
    document.getElementById("mySidenav").style.width = "0";
  }
  
  /* Loop through all dropdown buttons to toggle between hiding and showing its dropdown content - This allows the user to have multiple dropdowns without any conflict */
  var dropdown = document.getElementsByClassName("dropdown-btn");
  var i;
      
  for (i = 0; i < dropdown.length; i++) {
    dropdown[i].addEventListener("click", function() {  
      this.classList.toggle("active");
      var dropdownContent = this.nextElementSibling;
      if (dropdownContent.style.display === "block") {  
  
        dropdownContent.style.display = "none";
      } else {
        dropdownContent.style.display = "block";
      }
    });
  } 
  
  </script>

<div class="main">   

<br>

<h4>Bioinformatics tools</h4>

<p align="justify">Before we list the bioinformatics tools required to analyse genomic data, we 
will first need to understand various software management applications that we would need to run 
these tools.</p>

<br>

<h4>Software management applications</h4>

<p align="justify">Software management applications are indispensable tools in bioinformatics 
because they simplify software management, enhance reproducibility, support collaboration, and 
provide the flexibility needed to tackle the complex and dynamic nature of biological data 
analysis. These tools contribute to the efficiency and rigor of bioinformatics research and 
analysis workflows. Here, we will provide an overview of two software management 
applications, 
i.e., Docker and Conda.</p>


<h5>a. Docker</h5>

<p align="justify"; style="padding-left: 40px"><b>Overview</b></p>

<p align="justify"; style="padding-left: 40px">Docker is an open platform for developing, 
shipping, and running applications. It provides the ability to package and run an application in 
a loosely isolated environment called a container. Containers are lightweight and contain 
everything needed to run the application, so you do not need to rely on what is currently 
installed on the host; i.e., it involves bundling an application together with all of the 
necessary configuration files, libraries, and dependencies to ensure the software can run in a 
reproducible fashion across a diversity of computing environments. You can easily share 
containers while you work, and be sure that everyone you share with gets the same container that 
works in the same way.</p>

<p align="justify"; style="padding-left: 40px"><b>Docker architecture</b></p>

<p align="justify"; style="padding-left: 40px">Installing and running docker is dependent on the 
computer’s operating system. We will define some key concepts that you will come across when 
installing and running docker.</p>

<p align="center"><img width="600" height="340" alt="logo"
src="assets/images/logo/comp_3.png"></p>

<ul align="justify"; style="padding-left: 60px">
<li><b>Docker Desktop:</b> Docker Desktop is a user-friendly application for desktop and laptop 
computers that allows developers to create, manage, and run Docker containers on their local 
operating system. It simplifies the process of containerisation by providing a graphical 
interface and tools to build, test, and deploy applications within lightweight, isolated 
containers. Docker Desktop is commonly used for software development, testing, and application 
deployment, making it easier to work with containerised environments on personal computers.</li>

<li><b>Docker Engine:</b> Docker Engine is the core component of the Docker platform. It is 
responsible for creating and managing Docker containers, which are lightweight, portable, and 
self-contained environments for running applications.</li>

<li><b>Docker Daemon:</b> The Docker Daemon, often simply referred to as "the daemon," is a 
crucial component of the Docker platform. Docker daemon is the brain behind the whole operation - 
it plays a central role in managing Docker containers and orchestrating containerised 
applications.</li>

<li><b>Docker Client:</b> The Docker Client, often referred to as just "Docker," is a 
command-line tool or graphical user interface (GUI) that allows users to interact with the Docker Daemon and manage 
Docker containers, images, networks, volumes, and other Docker-related resources. It pipes/sends 
commands from your operating system terminal to the Docker environment. Docker client talks to 
Docker daemon.</li>
</ul>

<p align="justify"; style="padding-left: 40px">To install Docker on your system, you have two 
primary options: Docker Engine and Docker Desktop.</p>

<ul align="justify"; style="padding-left: 60px">

<li>If you prefer a lightweight Docker installation without additional graphical 
interfaces, Docker Engine is the choice to go with. Docker Engine can be installed on various 
Linux distributions.</li>

<li>Docker Desktop is a more user-friendly option that includes Docker Engine along with 
additional features and a graphical user interface. It's available for both Windows and 
macOS.</li>

</ul>

<p align="justify"; style="padding-left: 60px"><b><i>Note</p></b></i>

<p align="justify"; style="padding-left: 60px"><b><i>HyperKit</b> is a lightweight hypervisor 
designed for macOS operating 
systems. It is an 
open-source virtualisation tool that provides virtualisation capabilities on macOS, making it 
possible to run virtual machines (VMs) and containers on Mac computers. HyperKit is often 
associated with Docker for Mac, as it is one of the components used by Docker to enable 
containerisation and virtualisation on macOS.</I></p>

<p align="justify"; style="padding-left: 60px"><b><I>Hyper-V</b> is a virtualisation platform and 
hypervisor technology 
developed 
by Microsoft. It 
allows you to create and manage virtual machines (VMs) on Windows-based systems. Hyper-V is 
commonly used for server virtualisation, testing and development environments, and running 
multiple operating systems on a single physical machine.
</ul></i></p>

<p align="center"><img width="600" height="340" alt="logo"
src="assets/images/logo/comp_4.png"></p>


<p align="justify"; style="padding-left: 40px"><i><b>How to install 
Docker</b></i></p>

<p align="justify"; style="padding-left:60px"><i><b>a. Steps to install Docker on 
macOS:</i></b></p>

<p align="justify"; style="padding-left:80px"><i><b>1. Download Docker 
Desktop:</i></b></p>

<ul align="justify"; style="padding-left:100px">
<li><i>Go to the <a 
href="https://docs.docker.com/desktop/install/mac-install/">Docker Desktop for Mac 
page</a> on the Docker website.</i></li>

<li><i>Click the "Download from Docker 
Hub" button to download the Docker Desktop installer.</i></li>

</ul>

<p align="justify"; style="padding-left:80px"><i><b>2. Install Docker 
Desktop:</i></b></p>

<ul align="justify"; style="padding-left:100px">

<li><i>Open the downloaded <b>.dmg</b> 
file to mount the Docker Desktop disk image.</i></li>

<li><i>Drag the Docker icon to the 
Applications folder.</i></li>

</ul>

<p align="justify"; style="padding-left:80px"><i><b>3. Run Docker 
Desktop:</i></b></p>

<ul align="justify"; style="padding-left:100px">

<li><i>Open Docker Desktop from your 
Applications folder.</i></li>

</ul>

<p align="justify"; style="padding-left:80px"><i><b>4. Login to Docker Hub 
(Optional):</i></b></p>

<ul align="justify"; style="padding-left:100px">

<li><i>If you have a Docker Hub account, 
you can log in to Docker Desktop. This step is optional but allows you to access and 
download images from Docker Hub.</i></li>

</ul>

<p align="justify"; style="padding-left:80px"><i><b>5. Configure Resources 
(Optional):</i></b></p>

<ul align="justify"; style="padding-left:100px">

<li><i>In the Docker Desktop preferences, 
you can configure resources such as CPU, memory, and disk space allocated to Docker 
containers. Adjust these settings based on your system resources.</i></li>

</ul>

<p align="justify"; style="padding-left:80px"><i><b>6. Start Docker 
Desktop:</i></b></p>

<ul align="justify"; style="padding-left:100px">

<li><i>Click the Docker Desktop icon in 
your Applications folder to start Docker. The Docker icon in the menu bar indicates 
that Docker is running.</i></li>

</ul>

<p align="justify"; style="padding-left:80px"><i><b>7. Verify 
Installation:</i></b></p>

<ul align="justify"; style="padding-left:100px">

<li><i>To verify that Docker is installed 
and running, open a terminal and run the following commands:</i></li>

<p align="justify"; style="padding-left:80px"><code><b>docker 
--version</code></b></p>

<p align="justify"; style="padding-left:80px"><code><b>docker run 
hello-world</code></b></p>

<p align="center"><img width="400" height=“400” alt="logo"
src="assets/images/logo/comp_6.png"></p>

<br>
<li><i>The first command should display 
the Docker version, and the second command runs a simple container to verify that 
Docker can pull and run images.</i></li>

</ul>

<p align="justify"; style="padding-left:60px"><i><b>b. Steps to install Docker on 
ubuntu:</I></b></p>

<p align="justify"; style="padding-left:60px">Run the following commands on you 
ubuntu terminal</p>

<p align="justify"; style="padding-left:80px"><i><b>1. Update Package 
Lists:</i></b></p>

<p align="justify"; style="padding-left:80px"><code><b>sudo apt 
update</code></b></p>

<p align="justify"; style="padding-left:80px"><i><b>2. Install 
Prerequisites:</i></b></p>

<p align="justify"; style="padding-left:80px"><code><b>sudo apt install 
apt-transport-https ca-certificates curl software-properties-common</code></b></p>

<p align="justify"; style="padding-left:80px"><i><b>3. Add Docker's GPG 
Key:</i></b></p>

<p align="justify"; style="padding-left:80px"><code><b>curl -fsSL 
https://download.docker.com/linux/ubuntu/gpg | sudo gpg --dearmor -o 
/usr/share/keyrings/docker-archive-keyring.gpg</code></b></p>

<p align="justify"; style="padding-left:80px"><i><b>4. Set Up the Stable Docker 
Repository:</i></b></p>

<p align="justify"; style="padding-left:80px"><i>For Ubuntu 20.04 (Focal 
Fossa):</i></p>

<p align="justify"; style="padding-left:80px"><code><b>echo "deb [arch=amd64 
signed-by=/usr/share/keyrings/docker-archive-keyring.gpg] 
https://download.docker.com/linux/ubuntu focal stable" | sudo tee 
/etc/apt/sources.list.d/docker.list > /dev/null</code></b></p>

<p align="justify"; style="padding-left:80px"><i><b>5. Install Docker 
Engine:</i></b></p>

<p align="justify"; style="padding-left:80px"><code><b>sudo apt 
update</code></b></p>

<p align="justify"; style="padding-left:80px"><code><b>sudo apt install docker-ce 
docker-ce-cli containerd.io</code></b></p>

<p align="justify"; style="padding-left:80px"><i><b>6. Start and Enable 
Docker:</i></b></p>

<p align="justify"; style="padding-left:80px"><code><b>sudo systemctl start 
docker</code></b></p>

<p align="justify"; style="padding-left:80px"><code><b>sudo systemctl enable 
docker</code></b></p>

<p align="justify"; style="padding-left:80px"><i><b>7. Verify Docker 
Installation:</i></b></p>

<p align="justify"; style="padding-left:80px"><code><b>docker 
--version</code></b></p>

<p align="justify"; style="padding-left:80px"><i><b>8. Run a Test 
Container:</i></b></p>

<p align="justify"; style="padding-left:80px"><code><b>docker run 
hello-world</code></b></p>

<p align="justify"; style="padding-left:80px"><i>If everything is set up correctly, 
you should see a message indicating that your Docker installation is 
working.</i></p>

<p align="justify"; style="padding-left:80px"><i><b>9. Manage Docker as a Non-root 
User (Optional):</i></b></p>

<p align="justify"; style="padding-left:80px"><i>To avoid using <b>sudo</b> with 
every Docker command, add your user to the <b>docker</b> group:</i></p>

<p align="justify"; style="padding-left:80px"><code><b>sudo usermod -aG docker 
$USER</code></b></p>

<p align="justify"; style="padding-left:80px"><i>Log out and log back in or restart 
your system for the changes to take effect.</i></p>

<br>
<p align="justify"; style="padding-left: 40px"><i><b>Docker Usage</b></i></p>  

<p align="justify"; style="padding-left:60px"><i>In this course, we will be using 
Docker as a our software management application and Docker images (bioinformatics 
tools) built and catalogued by Sanger Pathogens, the State Public health 
Bioinformatics (StaPH-B) and NCBI.</i></p>

<p align="justify"; style="padding-left:80px"><i><b>Step 1: Downloading a Docker 
image</i></b></p>

<p align="justify"; style="padding-left:80px"><i>Download an image from a repository  
by using the <code><b>docker pull</b></code> command: </i></p>

<p align="justify"; style="padding-left:80px"><code><b>docker pull 
repository/(name_of_container):(tag)</b></code></p>

<p align="justify"; style="padding-left:80px"><i>For example</i></p>

<p align="justify"; style="padding-left:80px"><code><b>docker pull 
staphb/snippy</b></code></p>

<p align="justify"; style="font-size: 12px; padding-left:80px"><b><i>An explanation 
of 
this command is as
follows:</i></p></b>

<p align="justify"; style="font-size: 12px; padding-left:80px"><i><b>docker 
pull:</b> instructs Docker to download an image from a repository</I></p>

<p align="justify"; style="font-size: 12px; 
padding-left:80px"><i><b>staphb/snippy:</b> is the docker image</I></p>

<p align="justify"; style="padding-left:80px"><I>When you pull a docker image, you 
should see an output similar to this…</I></p>

<p align="center"><img width="400" height="180" alt="logo"
src="assets/images/logo/comp_7.png"></p>

<p align="justify"; style="padding-left:80px"><i><b>Step 2: Running a Docker
image</i></b></p>

<p align="justify"; style="padding-left:80px"><i>Starting a container from a docker 
image is simply done using the command <code><b>docker run</code></b>. A brand new 
Docker container is then created, and will run any command you provide within the 
container.</i></p>

<p align="justify"; style="padding-left:80px"><i>For example</i></p>

<p align="justify"; style="padding-left:80px"><code><b>docker run --rm=True -v 
$PWD:/data 
-u $(id -u):$(id -g) staphb/fastqc fastqc 21999_7#176*.fastq.gz 
</b></code></p>

<p align="justify"; style="font-size: 12px; padding-left:80px"><b><i>An explanation
of
this command is as
follows:</i></p></b>

<p align="justify"; style="font-size: 12px; padding-left:80px"><i><b>docker run:</b> 
function to start a new container </I></p>


<p align="justify"; style="font-size: 12px; padding-left:80px"><i><b>--rm=True:</b> 
By default, when a Docker container is run without this flag, the Docker container 
is created,the container runs, and then exits, but is not deleted. In other words, 
Docker containers are NOT ephemeral by default. A local copy of the container is 
kept and takes up unnecessary storage space. It is a good idea to always use this 
flag so that the container is removed after running it, unless for some reason you 
need the container after the specified program has been run.</I></p>

<p align="justify"; style="font-size: 12px; padding-left:80px"><i><b>-v 
$PWD:/data:</b> The <b>-v</b> flag mounts a volume between your local machine and 
the Docker container. This specific command mounts the present working directory to 
the /data directory within the Docker container, which makes the files on your local 
machine accessible to the container. You can change these paths to meet the needs of 
your system, however it is a good idea to have a working directory in each of the 
containers, and thus each container contains the /data directory for such 
purpose.</I></p>

<p align="justify"; style="font-size: 12px; padding-left:80px"><i><b>--u $(id 
-u):$(id -g):</b> By default, when Docker containers are run, they are run as the 
root user. This can be problematic because any files created from within the 
container will have root permissions/ownership and the local user will not be able 
to do much with them. The -u flag sets the container's user and group based on the 
user and group from the local machine, resulting in the correct file 
ownership.</I></p>


<p align="justify"; style="font-size: 12px;
padding-left:80px"><i><b>staphb/fastqc:</b> is the docker image. [ <b>staphb</b> - 
is the repository; <b>fastqc</b> - is the name of the container image]</I></p>

<p align="justify"; style="font-size: 12px;
padding-left:80px"><i><b>fastqc :</b> Command to run in that container</I></p>

<p align="justify"; style="font-size: 12px;
padding-left:80px"><i><b>21999_7#176*.fastq.gz:</b> Option(s) of the command (paths 
in options are based on container file system).</I></p>

<p align="justify"; style="padding-left:80px"><i>To avoid long commands, the [docker 
run --rm=True -v $PWD:/data -u $(id -u):$(id -g)] command can be incorporated into a 
bash function by including the following into your ~/.bashrc (if you are using Bash 
shell / Linux) or ~/.zshrc (if you are using Z shell / macOS)</i></p>

<p align="justify"; style="padding-left:80px"><i><b># For Bash Shell,</b> 
run the command:</i></p>

<p align="justify"; style="padding-left:80px"><i><b><code>echo 'function 
docker_run() { docker run --rm=True -u $(id -u):$(id -g) -v $(pwd):/data "$@" ;}' >> 
~/.bash_profile</code></b></i></p>

<p align="justify"; style="padding-left:80px"><i><b># For Z Shell,</b> run 
the command:</i></p>

<p align="justify"; style="padding-left:80px"><i><b><code>echo 'function 
docker_run() { docker run --rm=True -u $(id -u):$(id -g) -v $(pwd):/data "$@" ;}' >> 
~/.zshrc</b></i></code></p>

<p align="justify"; style="padding-left:80px"><i>You need to be in your home 
directory and then run the one of the following commands depending on your 
shell</i></p>

<p align="justify"; style="padding-left:80px"><i>You need to restart the Terminal 
for this change to take effect or Source the Bashrc by running the command:</i></p>

<p align="justify"; style="font-size: 12px;
padding-left:80px"><i><b><code>source ~/.bash_profile</b></code></i></p>

<p align="justify"; style="padding-left:80px"><i>To check the above command has 
worked properly open bashrc file using:
<code><b>cat .bash_profile</code></b> or <code><b>cat .zshrc</code></b></i></p>

<br>
<p align="justify"; style="padding-left: 40px"><i><b>Docker 
images/Bioinformatics tools</b></i></p>

<p align="justify"; style="padding-left:60px"><i>We will use various 
bioinformatics tools in this course, and the table below shows a list of 
tools required for this course. These tools can be accessed and executed 
using Docker platform.</i></p>

<br>

<style>
table, th, td {
  border: 1px solid black;
  border-collapse: collapse;
}
</style>

<table style="font-size: 12px; width:100%">
  <tr>
    <th>Tool</th>
    <th>Website</th> 
    <th>Brief Description</th>
    <th>Installation</th>
  </tr>
  <tr>
    <td>FastQC</td>
    <td>https://github.com/s-andrews/FastQC</td>
    <td>Quality assessment for high throughput sequencing datasets</td>
    <td>docker pull staphb/fastqc</td>
  </tr>
<tr>
    <td>Trimmomatic</td> 
    <td>https://github.com/usadellab/Trimmomatic</td>
    <td>Adapter trimming</td>
    <td>docker pull staphb/trimmomatic</td>
  </tr>
<tr>
    <td>bwa</td>
    <td>https://github.com/lh3/bwa</td>
    <td>Mapping DNA sequences</td>
    <td>docker pull staphb/bwa</td>
  </tr>
<tr>
    <td>samtools</td>
    <td>https://github.com/samtools/</td>
    <td>Manipulating NGS</td>
    <td>docker pull staphb/samtools</td>
  </tr>
<tr>
    <td>SPAdes</td>
    <td>https://github.com/ablab/spades/releases</td>
    <td><i>De novo</i> assembly</td>
    <td>docker pull staphb/spades</td>
  </tr>
<tr>
    <td>snp-sites</td>
    <td>https://github.com/sanger-pathogens/snp-sites</td>
    <td>Rapidly extract SNPs from a multi-FASTA alignment</td>
    <td>docker pull staphb/snp-sites</td>
  </tr>
<tr>
    <td>QUAST</td>
    <td>https://github.com/ablab/quast</td>
    <td>Quality assessment for genome assemblies</td>
    <td>docker pull staphb/quast</td>
  </tr>
<tr>
    <td>BLAST</td>
    <td>https://github.com/ncbi/blast_plus_docs</td>
    <td>Compares nucleotide or protein sequences to sequence databases and 
calculates the statistical significance of matches</td>
    <td>docker pull ncbi/blast</td>
  </tr>
<tr>
    <td>Kraken2</td>
    <td>https://github.com/DerrickWood/kraken2</td>
    <td>Kraken is an ultrafast and highly accurate program for assigning taxonomic 
labels to metagenomic DNA sequences</td>
    <td>docker pull staphb/kraken2</td>
  </tr>
<tr>
    <td>snippy</td>
    <td>https://github.com/tseemann/snippy</td>
    <td>Rapid haploid variant calling and core genome alignment</td>
    <td>docker pull staphb/snippy</td>
  </tr>
<tr>
    <td>FastTree</td>
    <td>http://www.microbesonline.org/fasttree/</td>
    <td>Phylogenetic tree builder</td>
    <td>docker pull staphb/fasttree</td>
  </tr>
<tr>
    <td>Gubbins</td>
    <td>https://github.com/nickjcroucher/gubbins</td>
    <td>Builds phylogeny after removing regions of recombination</td>
    <td>docker pull sangerpathogens/gubbins</td>
  </tr>
<tr>
    <td>popPUNK</td>
    <td>https://github.com/bacpop/PopPUNK</td>
    <td>Clustering</td>
    <td>docker pull staphb/poppunk</td>
  </tr>
<tr>
    <td>Abricate</td>
    <td>https://github.com/tseemann/abricate</td>
    <td>Mass screening of contigs for antimicrobial resistance or virulence 
genes</td>
    <td>docker pull staphb/abricate</td>
  </tr>
<tr>
    <td>Prokka</td>
    <td>https://github.com/tseemann/prokka</td>
    <td>Prokka is a software tool to annotate bacterial, archaeal and viral genomes 
quickly and produce standards-compliant output files</td>
    <td>docker pull staphb/prokka</td>
  </tr>
<tr>
    <td>Any2fasta</td>
    <td>https://github.com/tseemann/any2fasta</td>
    <td>Convert various sequence formats to FASTA</td>
    <td>docker pull staphb/any2fasta</td>
  </tr>
<tr>
    <td>ARIBA</td>
    <td>https://github.com/sanger-pathogens/ariba</td>
    <td>Antimicrobial Resistance Identification By Assembly</td>
    <td>docker pull staphb/ariba</td>
  </tr>
<tr>
    <td>SeroBA</td>
    <td>https://github.com/sanger-pathogens/seroba</td>
    <td>Identify the Serotype from Illumina NGS reads for given references</td>
    <td>docker pull staphb/seroba</td>
  </tr>
<tr>
    <td>SRST2</td>
    <td>https://github.com/katholt/srst2</td>
    <td>Gene detection and MLST</td>
    <td>docker pull staphb/srst2</td>
  </tr>
<tr>
    <td>mlst</td>
    <td>https://github.com/tseemann/mlst</td>
    <td>Scan contig files against traditional PubMLST typing schemes</td>
    <td>docker pull staphb/mlst</td>
  </tr>  
</table>

<br>
<p align="justify"; style="font-size: 12px; padding-left:80px"><i>To check that 
Docker images have been downloaded, run the command:</I></p>

<p align="justify"; style="padding-left:80px"><i><b><code>docker images | grep 
"staphb\|sanger\|ncbi"| wc -l</code></b></i></p>

<p align="justify"; style="font-size: 12px; padding-left:80px"><i>There should be 
20</i></p>

<br>
 
<h5>b. Conda</h5>

<p align="justify"; style="padding-left: 40px"><b>Definitions</b></p>

<p align="justify"; style="padding-left: 40px"><b>Anaconda:</b> is a distribution of 
Python and R programming languages for scientific computing, data science, machine 
learning, and related domains. It includes a variety of pre-installed libraries and 
tools commonly used in these fields. Anaconda simplifies the process of managing 
packages and environments for data science projects. Simply, Anaconda is a 
comprehensive distribution that includes a variety of pre-installed packages for 
data science and scientific computing.</p>

<p align="justify"; style="padding-left: 40px"><b>Miniconda:</b> is a minimal 
installer for the Conda package manager. Unlike Anaconda, which comes with a 
pre-selected set of packages, Miniconda allows users to install only the packages 
they need, making it a lightweight alternative. Simply, Miniconda is a minimal 
installer for Conda, allowing users to install only the packages they need for a 
more lightweight setup.</p>

<p align="justify"; style="padding-left: 40px"><b>Conda:</b> is the package manager 
used by both Anaconda and Miniconda. It is an open-source package management and 
environment management system that runs on Windows, macOS, and Linux. Simply, Conda 
is the package manager that can be used independently of Anaconda or Miniconda. It 
simplifies the process of managing packages and environments.</p>

<p align="justify"; style="padding-left: 40px"><b>Conda environment:</b> A Conda 
environment is a self-contained directory that contains a specific collection of 
Python (or other programming language) packages and their dependencies. Environments 
allow you to manage and isolate different sets of packages, ensuring that your 
projects have the specific dependencies they need without interfering with each 
other.</p>

<p align="justify"; style="padding-left:60px"><i><b>i. Steps to install 
Anaconda:</I></b></p>

<p align="justify"; style="padding-left:80px"><i><b>1. Download:</i></b></p>

<ul align="justify"; style="padding-left:100px">
<li><i>Visit the <a
href="https://www.anaconda.com/download">Anaconda download page</a> and choose the 
appropriate version (Python 3.x is recommended).</i></li>

</ul>

<p align="justify"; style="padding-left:80px"><i><b>2. Install:</i></b></p>

<ul align="justify"; style="padding-left:100px">
<li><i>Follow the installation instructions for your operating system. The 
installation typically involves running an installer and accepting the license 
terms.</i></li>

</ul>

<p align="justify"; style="padding-left:80px"><i><b>3. Verify 
Installation:</i></b></p>

<ul align="justify"; style="padding-left:100px">

<li><i>After installation, you can open a terminal or command prompt and 
run:</i></li>

<p align="justify"; style="padding-left:80px"><code><b>conda 
--version</code></b></p>

<li><i>This command should display the installed version of conda.</i></li>

</ul>

<p align="justify"; style="padding-left:80px"><i><b>4. Create and Activate an 
Environment (Optional):</i></b></p>

<ul align="justify"; style="padding-left:100px">

<li><i>You can create a new environment using:</i></li>

<p align="justify"; style="padding-left:80px"><code><b>conda create --name 
myenv</code></b></p>

<i>Replace "myenv" with the desired name for your environment.</i>

<li><i>Activate the environment:</i></li>

<p align="justify"; style="padding-left:80px"><code><b>conda activate 
myenv</code></b></p>

</ul>

<p align="justify"; style="padding-left:60px"><i><b>ii. Steps to install 
Miniconda:</I></b></p>

<p align="justify"; style="padding-left:80px"><i><b>1. Download and 
Install:</I></b></p>

<ul align="justify"; style="padding-left:100px">
<li><i>Visit the <a
href="https://docs.conda.io/projects/miniconda/en/latest/">Miniconda website 
page</a> and then follow the installation instructions.</i></li>

</ul>

<p align="justify"; style="padding-left:80px"><i><b>2. Create and Activate an 
Environment (Optional):</i></b></p>

<ul align="justify"; style="padding-left:100px">

<li><i>You can create a new environment using:</i></li>

<p align="justify"; style="padding-left:80px"><code><b>conda create --name 
myenv</code></b></p>

<i>Replace "myenv" with the desired name for your environment.</i>

<li><i>Activate the environment:</i></li>

<p align="justify"; style="padding-left:80px"><code><b>conda activate 
myenv</code></b></p>

</ul>

<p align="justify"; style="padding-left:80px"><i><b>3. Install Packages:</i></b></p>

<ul align="justify"; style="padding-left:100px">

<li><i>Once the environment is activated, you can use conda to install packages. For 
example:</i></li>

<p align="justify"; style="padding-left:80px"><code><b>conda install 
numpy</code></b></p>

</ul>

<p align="justify"; style="padding-left:80px"><i><b>3. Deactivate the 
Environment:</i></b></p>

<ul align="justify"; style="padding-left:100px">

<li><i>When you're done working in the environment, you can deactivate it:</i></li>

<p align="justify"; style="padding-left:80px"><code><b>conda 
deactivate</code></b></p>

</ul>

<p align="justify"; style="padding-left:80px"><i><b>4. Remove an Environment 
(Optional):</i></b></p>

<ul align="justify"; style="padding-left:100px">

<li><i>If you want to remove an environment, you can use:</i></li>

<p align="justify"; style="padding-left:80px"><code><b>conda env remove --name 
myenv</code></b></p>

</ul>

<p align="justify"; style="padding-left:80px"><i><b>N/B:</B>Miniconda is a great 
choice for users who prefer a more minimalistic and customizable setup. It allows 
you to build environments tailored to specific projects or use cases without the 
overhead of a larger distribution like Anaconda.</I></p>
  
 </div>
    

</body>
</html>


