---
layout: home
# articles:
#   excerpt_type: html
---
<!DOCTYPE html>
<html>
<head>
<meta name="viewport" content="width=device-width, initial-scale=1">
<style>
body {
  font-family: "Lato", sans-serif;
}

.sidenav {
  height: 100%;
  width: 400px;
  position: fixed;
  z-index: 1;
  top: 0;
  right: 0;
  background-color: #0C6980;
  overflow-x: hidden;
  padding-top: 40px;
  transition: 0.5s;
}

.sidenav a, .dropdown-btn {
  padding: 10px 20px 10px 10px;
  text-decoration: none;
  font-size: 12px;
  color: white;
  display: block;
  cursor: pointer; 
  outline: none;
  border: none;
  background: none;  
  width: 100%;  
  text-align: right;
  transition: 0.3s;
 

}

.sidenav a:hover, .dropdown-btn:hover {
  color: #1e504f;
}

.sidenav .closebtn {
  position: absolute;
  top: 0;
  right: 0px;
  font-size: 20px;
  right: 0px;
}

.openbtn {
  font-size: 20px;
  cursor: pointer;
  background-color: #7cc7bb;
  color: white;
  padding: 10px 15px;
  border: none;
}

.main {
  margin-right: 450px; /* Same as the width of the sidenav */
  font-size: 14px; /* Increased text to enable scrolling */
  padding: 0px 10px;
}

.active {  
  background-color: #77A7B0;
  color: white;
} 

.dropdown-container {
  display: none;
  background-color: #7cc7bb;
  padding-left: 8px;
}

.fa-caret-down {
  float: left;
  padding-left: 8px;
}


@media screen and (max-height: 450px) {
  .sidenav {padding-top: 15px;}
  .sidenav a {font-size: 18px;}
}
</style>
</head>
<body>

<br>

<b><button class="openbtn"; style="font-size:14px;cursor:pointer;font-family:sans-serif" onclick="openNav()";>☰ Content</button></b> 


<div id="mySidenav" class="sidenav">
  <a href="javascript:void(0)" class="closebtn" onclick="closeNav()">&times;</a>
  <a href="Bioinformatics_Workshop.html">Course Overview</a>

  <button class="dropdown-btn">Data, Tools and Computational Platforms
    <i class="fa fa-caret-down"></i>
  </button>
  <div class="dropdown-container">
    <a href="computing.html">Overview</a>
    <a href="comp_specs.html">Computer Specifications</a>
    <a href="docker.html">Software Management Apps (Docker) & Bioinformatics Tools</a>
  </div>
  
<button class="dropdown-btn">Genomic Data Analysis (Bioinformatics)
    <i class="fa fa-caret-down"></i>
  </button>
  <div class="dropdown-container">
    <a href="analysis.html">Overview</a>
    <a href="ngs_file_formats.html">NGS File Formats</a>
    <a href="ngs_data_QC.html">NGS Data Quality Control</a>
    <a href="adapter_trimming.html">Adapter Trimming</a>
    <a href="taxonomy.html">Taxonomy Sequence Classification</a>
    <a href="denovo_assembly.html">de novo Assembly</a>
    <a href="quality_assessment_genome_assemblies.html">Quality Assessment
of Genome Assemblies</a>
    <a href="genome_annotation.html">Genome Annotation</a>
    <a href="serotyping.html">Serotyping</a>
    <a href="mlst.html">Multilocus Sequence Typing</a>
    <a href="amr.html">AMR Profiling</a>
    <a href="mapping.html">Mapping</a>
    <a href="variant_calling.html">Variant Calling</a>
    <a href="phylogenetics.html">Phylogenetics</a>
  </div>   
  
<a href="trainers.html">Acknowledgement</a>
  
</div>

<script>
  function openNav() {
    document.getElementById("mySidenav").style.width = "400px";
  }
  
  function closeNav() {
    document.getElementById("mySidenav").style.width = "0";
  }
  
  /* Loop through all dropdown buttons to toggle between hiding and showing its dropdown content - This allows the user to have multiple dropdowns without any conflict */
  var dropdown = document.getElementsByClassName("dropdown-btn");
  var i;
      
  for (i = 0; i < dropdown.length; i++) {
    dropdown[i].addEventListener("click", function() {  
      this.classList.toggle("active");
      var dropdownContent = this.nextElementSibling;
      if (dropdownContent.style.display === "block") {  
  
        dropdownContent.style.display = "none";
      } else {
        dropdownContent.style.display = "block";
      }
    });
  } 
  
  </script>

<div class="main">   
<br>
<br>
<p align="left"><i>Before you begin this section, download this folder <a 
href="https://drive.google.com/drive/folders/1X0ZZw7yvtlqWcf3jDfNSqr94bsp-u2qN?usp=share_link">QC</a>. 
You will use this folder and its contents to learn and practice this section.</i></p>
<br>

<h4>Overview</h4>

<p align="left">A typical whole genome sequencing process involves sample preparation, 
library preparation and sequencing. Errors occurring at each of these steps can negatively 
impact the quality of the sequence information. For example, the samples could be mixed 
during sample or library preparation, or errors can be encountered during the sequencing 
itself. If these errors are not removed from the raw reads, they might be incorporated into 
your analysis output and would be harder to resolve later on. Therefore, it is important to 
perform quality checks on the raw sequence reads before starting your analysis.
<br>
<br>
Here are some of the parameters for NGS data quality control.
<ul style="padding-left: 30px">
  <li>Check for primers and adaptors</li>
  <li>Trim low quality ends</li>
  <li>Remove low quality reads</li>
  <li>Remove short sequences</li>
  <li>Remove reads with a high proportion of ambiguous bases (“N”)</li>
  <li>Remove duplicates</li>
</ul>
<br>
There are several bioinformatic tools available for evaluating read data quality, here we will 
discuss one of the widely used tools, called FastQC Links to an external site. which is made 
available by the Babraham Institute. The tool also has a graphical user interface (GUI) so can be 
used without using the command line. The GUI aspect of this tool has been described in the <a 
href="https://docs.google.com/document/d/1mn2OGqFxk_rHotxST2CLlDv7FIJyCxO_xES69DZX47w/edit#">F2 
module</a>.
<br>
<br>
In this course, we will be using the command line. 
 </p>

<h4>Tool(s)</h4>
<p align="left">We will be using FastQC tool to carry out quality control on reads. If you do not 
have this tool in your local machine, you can download them from a docker repository using the 
command: <code><b>docker pull staphb/fastqc</code></b></p>

<p align="center"><img width="400" height="240" alt="logo"
src="assets/images/logo/qc_1.png"></p>

<p align="center"><img width="400" height="240" alt="logo"
src="assets/images/logo/qc_2.png"></p>

<h4>Quality Control (QC)</h4>

<ol>
  <li><b>Download data to be analysed using the FastQC</b></li>
<p>We will be using the reads (<b>21999_7#176_1.fastq.gz</b> and <b>21999_7#176_2.fastq.gz</b>) 
to learn the 
FastQC tool.</p>
<p>We will navigate to the folder containing these fastqs: <code><b>cd QC</code></b></p>

<p align="center"><img width="400" height="240" alt="logo"
src="assets/images/logo/qc_3.png"></p>

  <li><b>Execute Fastqc</b></li>

<p>To execute the FastQC program on the reads (<b>21999_7#176_1.fastq.gz</b> and
<b>21999_7#176_2.fastq.gz</b>), run the command: <code><b>docker_run staphb/fastqc fastqc 
21999_7#176*.fastq.gz</code></b></p>

<p align="center"><img width="400" height="240" alt="logo"
src="assets/images/logo/qc_4.png"></p>

<p style="font-size: 15px; padding-left: 30px"><b>An explanation of this command is as 
follows:</b></p>
<p style="font-size: 15px; padding-left: 30px"><b>docker_run:</b>is a function to start a 
container. The function includes the 
following flags: docker run --rm=True -u $(id -u):$(id -g) -v $(pwd):/data "$@". To understand 
the docker_run function read the section [Data, Tools and Computational Platforms (Docker)]</p>
<p style="font-size: 15px; padding-left: 30px"><b>staphb/fastqc_v0.11.9:</b> is the docker image 
[<b>staphb</b> - represents 
the repository and <b>fastqc_v0.11.9</b> - represents the container image]</p>
<p style="font-size: 15px; padding-left: 30px"><b>fastqc:</b> is the tool</p>
<p style="font-size: 15px; padding-left: 30px"><b>21999_7#176*.fastq.gz:</b> input files;  the 
(*) sign tells fastqc tool to 
run on files that start with 21999_7#176 and end with fastq.gz in the folder</p> 
</ol> 

You will have the following output 

<p align="center"><img width="400" height="240" alt="logo"
src="assets/images/logo/qc_5.png"></p>

When you <code><b>ls</code></b> you current directory (QC), the QC folder will now have the 
following files

<p align="center"><img width="400" height="240" alt="logo"
src="assets/images/logo/qc_6.png"></p>


<p align="left">Let’s explore the <b>21999_7#176_1_fastqc.html</b></p>

<h5>Basic Statistics</h5>

<p align="center"><img width="400" height="240" alt="logo"
src="assets/images/logo/qc_7.png"></p>

<p>Looking at the HTML file for 21999_7#176_1.fastq.gz, the basic statistics are shown in the 
figure above. The %GC which is 39 as should be for S. pneumoniae. There are 0 reads assigned as 
bad quality indicating good quality sequence data.</p> 

<p>The Basic Statistics module generates some simple composition statistics for the file 
analysed.</p> 

<p><b>Filename:</b> The original filename of the file which was analysed</p>

<p><b>File type:</b> Says whether the file appeared to contain actual base calls or colorspace 
data which 
had to be converted to base calls.</p>

<p><b>Encoding:</b> Says which ASCII encoding of quality values was found in this file.</p>

<p><b>Total Sequences:</b> A count of the total number of sequences processed. There are two 
values 
reported, actual and estimated. At the moment these will always be the same. In the future it may 
be possible to analyse just a subset of sequences and estimate the total number, to speed up the 
analysis, but since we have found that problematic sequences are not evenly distributed through a 
file we have disabled this for now.</p>

<p><b>Sequence Length:</b> Provides the length of the shortest and longest sequence in the set. 
If all sequences are the same length only one value is reported.</p> 

<p><b>%GC:</b> The overall %GC of all bases in all sequences.</p>

<h5>Per base sequence quality</h5>

<p align="center"><img width="400" height="240" alt="logo"
src="assets/images/logo/qc_8.png"></p>


<p>Scrolling down the HTML file is the ‘per base sequence quality’ figure, shown above. The graph 
represents the average quality of bases over the entire read length. The y-axis on the graph 
shows the quality scores. The space coloured in the green regions shows high quality, the amber 
coloured region below reflects acceptable quality and the regions in the red shows low quality. 
Therefore, if you observe the line in the red region for your sequence reads, it means the 
probability of a base call being incorrect is higher.</p>

<p>For each position a BoxWhisker type plot is drawn. The elements of the plot are as follows:
<ul style=“padding-left: 30px”>
<li>The central red line is the median value</li>
<li>The yellow box represents the inter-quartile range (25 - 75)</li>
<li>The upper and lower whiskers represent the 10</li>
<li>The blue line represents the mean quality</li>
</ul>
<p>The quality of calls on most platforms will degrade as the run progresses, so it is common to 
see base calls falling into the orange areas towards the end of a read.</p>
<p>It should be mentioned that there are a number of different ways to encode a quality score in 
a FastQ file. FastQC attempts to automatically determine which encoding method was used, but in 
some very limited datasets it is possible that it will guess this incorrectly (ironically only 
when your data is universally very good!). The title of the graph will describe the encoding 
FatsQC thinks your file used.</p>

<h5>Per tile sequence quality</h5>

<p align="center"><img width="400" height="240" alt="logo"
src="assets/images/logo/qc_9.png"></p>

<p>This graph will only appear in your analysis results if you’re using an illumina library which 
retains its original sequence identifiers. Encoded in these is the flowcell tile from which each 
read came. The graph allows you to look at the quality scores from each tile across all of your 
bases to see if there was a loss in quality associated with only one part of the flow cell.</p>
<p>The plot shows the deviation from the average quality for each tile. The colours are on a cold 
to hot scale, with cold colours being positions where the quality was at or above the average for 
that base in the run, and hotter colours indicate that a tile had worse qualities than other 
tiles for that base. A good plot should be blue all over.</p>
<p>Reasons for seeing warnings or errors on this plot could be transient problems such as bubbles 
going through the flowcell, or they could be more permanent problems such as smudges on the 
flowcell or debris inside the flowcell lane.</p>

<h5>Per Sequence Quality Scores</h5>

<p align="center"><img width="400" height="240" alt="logo"
src="assets/images/logo/qc_10.png"></p>

<p>The per sequence quality score report allows you to see if a subset of your sequences have 
universally low quality values. It is often the case that a subset of sequences will have 
universally poor quality, often because they are poorly imaged (on the edge of the field of view 
etc), however these should represent only a small percentage of the total sequences.</p>
<p>If a significant proportion of the sequences in a run have overall low quality then this could 
indicate some kind of systematics problem - possible with just part of the run (for example one 
end of a flowcell).</p>

<h5>Per Base Sequence Content</h5>

<p align="center"><img width="400" height="240" alt="logo"
src="assets/images/logo/qc_11.png"></p>

<p>Per Base Sequence Content plots out the proportion of each base position in a file for which 
each of the four normal DNA bases has been called.</p>
<p>In a random library you would expect that there would be little to no difference between the 
different bases of a sequence run, so the lines in this plot should run parallel with each other, 
The relative amount of each base should reflect the overall amount of these bases in your genome, 
but in any case they should not be hugely imbalanced from each other.</p>
<p>It’s worth noting that some types of library will always produce biased sequence composition, 
normally at the start of the read. Libraries produced by priming using random hexamers (including 
nearly all RNA-seq libraries) and those which were fragmented using transposases inherit an 
intrinsic bias in the positions at which reads start. This bias does not concern an absolute 
sequence, but instead provides enrichment of a number of different k-mers at the 5’end of the 
reads. Whilst this is a true technical bias, it isn't something which can be corrected by 
trimming and in most cases does not seem to adversely affect the downstream analysis. It will 
however produce a warning or error in this module.</p>

<h5>Per Sequence GC Content</h5>

<p align="center"><img width="400" height="240" alt="logo"
src="assets/images/logo/qc_12.png"></p>

<p>This module measures the GC content across the whole length of each sequence in a file and 
compares it to a modelled normal distribution of GC content.</p>
<p>In a normal random library you would expect to see a roughly normal distribution of GC content  
where the central peak corresponds to the overall GC content of the underlying genome. Since we 
don't know the GC content of the genome the modal GC content miscalculated from the observed data 
and used to build a reference distribution.</p>
<p>An unusually shaped distribution could indicate a contaminated library or some other kind of 
biased subset. A normal distribution which is shifted indicates some systematic bias which is 
independent of base position. If there is a systematics bias which creates a shifted normal 
distribution then this won't be flagged as an error by the module since it doesn't know what your 
genome’s GC content.</p>
 
<h5>Per Base N Content</h5>

<p align="center"><img width="400" height="240" alt="logo"
src="assets/images/logo/qc_13.png"></p>

<p>If a sequencer is unable to make a base call with sufficient confidence then it will normally 
substitute an N rather than a conventional base call.</p>
<p>This module plots out the percentage of base calls at each position for which an N was 
called.</p>
<p>It's not unusual to see a very low proportion of Ns appearing in a sequence especially nearer 
the end of a sequence. However, if this proportion rises above a few percent it suggests that the 
analysis pipeline was unable to interpret the data well enough to make valid base calls.</p>

<h5>Sequence Length Distribution</h5>

<p align="center"><img width="400" height="240" alt="logo"
src="assets/images/logo/qc_14.png"></p>

<p>Some high throughput sequencers generate fragments of uniform length but others can contain 
reads of wildly varying lengths. Even with uniform length libraries, some pipelines will trim 
sequences to remove poor quality base calls from the end.</p>
<p>This module generates a graph showing the distribution of fragment sizes in the file which was 
analysed.</p>
<p>In many cases this will produce a simple graph showing a peak only at one size, but for 
variable length FastQ files this will show the relative amounts of each different size of 
sequence fragment.</p>

<h5>Sequence Duplication Levels</h5>

<p align="center"><img width="400" height="240" alt="logo"
src="assets/images/logo/qc_15.png"></p>

<p>In a diverse library most sequences will occur only once in the final set. A low level of 
duplicating may indicate a very high level of coverage of the target sequence, but a high level 
of duplication is more likely to indicate some kind of enrichment bias (e.g., PCR over 
amplification).</p>
<p>This module counts the degree of duplication for every sequence in a library and creates a 
plot showing the relative number of sequences with different degrees of duplications.</p>
<p>To cut down on the memory requirements for this module only sequences which first appear in 
the first 100,000 sequences in each file are analysed, but this should be enough to get a good 
impression for the duplication levels in the whole file. Each sequence is tracked to the end of 
the file to give a representative count of the overall duplication level. To cut down on the 
amount of information in the final plot any sequence with more than 10 duplicates are placed into 
grouped bins to give a clear impression of the overall duplication level without having to show 
each individual duplication value.</p>
<p>Because the duplication detection requires an exact sequence match over the whole length of 
the sequence, any reads over 75bps in length are truncated to 50bp for the purposes of this 
analysis. Even so, longer reads are more likely to contain sequencing errors which will 
artificially increase the observed diversity and will tend to underrepresent highly duplicated 
sequences.</p>
<p>The plot shows the proportion of the library which is made up of sequences in each of the 
different duplication level bins. There are two lines in the plot. The  blue line takes the full 
sequence set and shows how its duplication levels are distributed. In the red plot, the sequences 
are de-duplicated and the proportions shown are the proportions of the de-duplicated set which 
come from different duplication levels in the original data.</p>
<p>In a properly diverse library, most sequences should fall into the far left of the plot in 
both the red and blue limes. A general level of enrichment, indicating broad over sequencing in 
the library will tend to flatten the lines, lowering the low end and generally raising other 
categories. More specific enrichments of subsets, or the presence of low complexity contaminants 
will tend to produce spikes towards the right of the plot. These high duplication peaks will most 
often appear on the blue trace as they make up a high proportion of the original library, but 
usually disappear in the read trace as they make up an insignificant proportion of the 
deduplicated set. If peaks persist in the blue trace, then this suggests that there are a large 
number of different highly duplicated sequences which might indicate either a contaminant set or 
a very severe technical duplication.</p>
<p>The module also calculates an expected overall loss of sequence were the library to be 
duplicated. This headline figure is shown at the top of the plot and gives a reasonable 
impression of the potential overall level of loss.</p> 

<h5>Overrepresented sequences</h5>

<p align="center"><img width="400" height="100" alt="logo"
src="assets/images/logo/qc_16.png"></p>

<p>A normal high-throughput library will contain a diverse set of sequences, with no individual 
sequence making up a tiny fraction of the whole. Finding that a single sequence is very 
over-represented in the set either means that it is highly biologically significant, or indicates 
that the library is contaminated , or not as diverse as you expected.</p>
<p>This module lists all of the sequences which make up more than 0.1.</p>
<p>For each overrepresented sequence the program will look for matches in a database of common 
contaminants and will report the hit it finds. Hits must be at least 20bp in length and have no 
more than 1 mismatch. Finding a hit doesn't necessarily mean that this is the source of the 
contamination, but may point you in the right direction. It's also worth pointing out that many 
adapter sequences are very similar to each other so you may get a hit reported which isn't 
technically correct, but which has a very similar sequence to the actual match.</p>

<h5>Sequence Duplication Levels</h5>

<p align="center"><img width="400" height="240" alt="logo"
src="assets/images/logo/qc_17.png"></p>

<p>The k-mer content module will do a generic analysis of all of the k-mers in your library to 
find those which do not have even coverage through the length of your reads. This can find a 
number of different sources of bias in the library which can include the presence of read-through 
adapter sequences building up  the end of your sequences.</p>
<p>You can however find that the presence of any overrepresented sequences in your library (such 
as adapter dimers) will cause the k-mer plot to be dominated by the k-mers these sequences 
contain, and that it's not always easy to see if there are other biases present in which you 
might be interested.</p>
<p>One obvious class of sequences which you might want to analyse are adapter sequences. It is 
useful to know if your library contains a significant amount of adapter in order to be able to 
assess whether you need to trim the adapter or not. Although the k-mer analysis can theoretically 
spot this kind of contamination it isn't always clear. This module therefore does a specific 
search for a set of separately defined k-mers and will give you a view of the total proportion of 
your library which contains these k-mers. A results trace will always be generated for all the 
sequences present in the adapter config file so you can see the adapter content of your library, 
even if it is low.</p>
<p>The plot itself shows a cumulative percentage count of the proportion of your library which 
has seen each of the adapter sequences at each position. Once a sequence has been seen in a read, 
it is counted as being present right through to the end of the read so the percentages you see 
will only increase as the read length goes on.</p>

<p>Here is the summary of all the statistics of all the statistics according to developer’s 
page:<a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/"> fastqc</a></p>
  
 </div>
    

</body>
</html>

