---
layout: home
# articles:
#   excerpt_type: html
---
  
  
<!DOCTYPE html>   
<html>  
<head>  
<meta name="viewport" content="width=device-width, initial-scale=1">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">
<style>
body {
  font-family: "Lato", sans-serif;
}
  
/* Fixed sidenav, full height */
.sidenav {
  height: 90%;
  width: 400px;
  position: fixed;
  z-index: 1;
  top: 40;
  left: 0;
  background-color: #ffffff00;
  overflow-x: hidden;
  padding-top: 40px;
}
  
/* Style the sidenav links and the dropdown button */
.sidenav a, .dropdown-btn {
  padding: 6px 8px 6px 16px;
  text-decoration: none;
  font-size: 12px;
  color: 
#386E90;
  display: block;
  border: none;
  background: none;  
  width: 100%;  
  text-align: left;
  cursor: pointer;  
  outline: none;
}

/* On mouse-over */
.sidenav a:hover, .dropdown-btn:hover {
  color:
#87B1B0;
} 

/* Main content */
.main {
  margin-left: 200px; /* Same as the width of the sidenav */
  font-size: 14px; /* Increased text to enable scrolling */
  padding: 0px 10px;
}

/* Add an active class to the active dropdown button */

.active {  
  background-color: #77A7B0;
  color: white;
} 

/* Dropdown container (hidden by default). Optional: add a lighter background color and some left padding to change the design of the dropdown content */
.dropdown-container {
  display: none;
  background-color: #E5F2F0;
  padding-left: 8px;
}

/* Optional: Style the caret down icon */
.fa-caret-down {
  float: right;
  padding-right: 8px;
}
  
/* Some media queries for responsiveness */
@media screen and (max-height: 450px) {
  .sidenav {padding-top: 15px;}
  .sidenav a {font-size: 18px;}
}
</style>
</head>
<body>
 
<div class="sidenav">
  <a href="Bioinformatics_Workshop.html">Course Overview</a>

<button class="dropdown-btn">Data, Tools and Computational Platforms
    <i class="fa fa-caret-down"></i>
  </button>
  <div class="dropdown-container">
    <a href="computing.html">Overview</a>
    <a href="comp_specs.html">Computer Specifications</a>
    <a href="docker.html">Software Management Apps (Docker) & Bioinformatics Tools</a>
  </div>
  
<button class="dropdown-btn">Genomic Data Analysis (Bioinformatics)
    <i class="fa fa-caret-down"></i>
  </button>
  <div class="dropdown-container">
    <a href="analysis.html">Overview</a>
    <a href="ngs_file_formats.html">NGS File Formats</a>
    <a href="ngs_data_QC.html">NGS Data Quality Control</a>
    <a href="adapter_trimming.html">Adapter Trimming</a>
    <a href="taxonomy.html">Taxonomy Sequence Classification</a>
    <a href="denovo_assembly.html">de novo Assembly</a>
    <a href="quality_assessment_genome_assemblies.html">Quality Assessment
of Genome Assemblies</a>
    <a href="genome_annotation.html">Genome Annotation</a>
    <a href="serotyping.html">Serotyping</a>
    <a href="mlst.html">Multilocus Sequence Typing</a>
    <a href="amr.html">AMR Profiling</a>
    <a href="mapping.html">Mapping</a>
    <a href="variant_calling.html">Variant Calling</a>
    <a href="phylogenetics.html">Phylogenetics</a>
  </div>   
  
<a href="trainers.html">Acknowledgement</a>
  
</div>

<div class="main">   
<br>
<br>
    
<h4>Overview</h4>
<p align="justify">Phylogenetics is the study of evolutionary relationships among biological entities – 
often species, individuals or genes (which may be referred to as taxa). The major elements of 
phylogenetics are summarised in figure below.</p>

<p align="center"><img width="400" height="400" alt="logo"
src="assets/images/logo/phylo_1.png"></p>
<p align="center"><a 
href="https://www.ebi.ac.uk/training/online/courses/introduction-to-phylogenetics/what-is-phylogenetics/">What 
is phylogenetics?</a>

<p align="justify">Phylogenetic trees based on whole genome data tell us about the relationships of 
bacterial isolates to each other on a very fine scale. When we combine that high-resolution information 
about the evolutionary relationships of isolates with geographical data it can inform our understanding of 
the current distribution of the pathogen and allow us to infer the epidemiological processes that have 
acted on the bacteria over time. The simplest example of this would be if a phylogeny showed that a 
pathogen was geographically constrained (e.g. isolates from the same region always cluster together). This 
might indicate that the pathogen is not rapidly spreading. Whereas a pathogen with a phylogeny that shows 
isolates from distant regions are likely to be related to isolates from nearby, the interpretation is that 
the pathogen is likely to be spread across regional borders. Geographical referencing of genomic data can 
also be combined with temporal information to study the movement of pathogens in space and time. This is 
most useful when done in real time and thus can be useful for outbreak detection and monitoring.</p>

<p align="justify">In this module we will cover: (i) building phylogenetic trees, (ii)identifying regions of 
recombination and (iii)clustering using popPUNK.</p>

<br>

<h5>Creating a phylogenetic tree using FastTree</h5>

<p align=“justify”><I>You will use the <b>variant calling</b> results to learn and practice this section.</p></I>

<h4>Tool(s)</h4>
<p align="justify">In this section, we will be using the following tools: snp_sites and FastTree tools - you  can download these tools  from a docker 
repository using 
the commands:</p>

<code><b>docker pull staphb/snp-sites</code></b>
<br>
<code><b>docker pull staphb/fasttree</code></b>
<br>

<h4>Make a SNP-only alignment using snp-sites</h4>
<p align="justify">Creating a phylogeny from whole genome sequences can be a very time consuming and 
computationally intensive process. We can speed this up by only using the variable sites (SNPs). However, 
we need to be aware that only including variable sites can affect the evolutionary rate estimates made by 
phylogenetics software - therefore, we need to account for the sites we remove in our analysis.</p>

<p align="justify">We will use snp-sites to do this. You can view the options for snp-sites using the 
command:</p>

<code><b>docker run staphb/snp-sites snp-sites -h</code></b>
<br>

<p align="center"><img width="400" height="240" alt="logo"
  src="assets/images/logo/phylo_2.png"></p>

<p align="justify">First, remove all the invariant sites and create a SNP-only multiple sequence 
alignment. We will use output from snippy runs described in the previous page. Run the command: </p>

<code><b>docker_run staphb/snp-sites snp-sites -o clean.full.SNPs.aln clean.full.aln</code></b>
<br>

<p style="font-size: 12px; padding-left: 30px"><b>An explanation of these commands are as follows:</b></p>
<p style="font-size: 12px; padding-left: 30px"><b>docker_run:</b> is a function to start a container. The 
function includes the following flags: docker run --rm=True -u $(id -u):$(id -g) -v $(pwd):/data "$@". To 
understand the docker_run function read the section [Data, Tools and Computational Platforms (Docker)]</p>
<p style="font-size: 12px; padding-left: 30px"><b> staphb/snp-sites:</b> is the docker image</p>
<p style="font-size: 12px; padding-left: 30px"><b> snp-sites:</b> is the tool</p>
<p style="font-size: 12px; padding-left: 30px"><b>-o clean.full.SNPs.aln:</b> specifies the output 
file</p>
<p style="font-size: 12px; padding-left: 30px"><b>clean.full.aln:</b> specifies the input file - which is 
an output of snippy </p>

<p align="justify">We can see how many invariant sites were removed (and what proportion of A, T, G, C 
they were) using: </p>

<code><b>docker_run staphb/snp-sites snp-sites -C clean.full.aln</code></b>
<br>

<p align="center"><img width="400" height="240" alt="logo"
  src="assets/images/logo/phylo_3.png"></p>

<h4>Creating a phylogenetic tree from the SNPs using FastTree</h4>

<p align="justify">You can view options for fasttree as follows:</p>

<code><b>docker_run staphb/fasttree FastTree FastTree -h</code></b>
<br>


<p align="center"><img width="400" height="240" alt="logo"
  src="assets/images/logo/phylo_4.png"></p>

<p align="justify">We will generate a maximum likelihood phylogenetic tree using this command: </p>

<code><b>docker_run staphb/fasttree FastTree -nt -gtr clean.full.SNPs.aln > 
clean.full.SNPs.aln.tree</code></b>
<br>

<p style="font-size: 12px; padding-left: 30px"><b>An explanation of these commands are as follows:</b></p>
<p style="font-size: 12px; padding-left: 30px"><b>docker_run:</b> is a function to start a container. The 
function includes the following flags: docker run --rm=True -u $(id -u):$(id -g) -v $(pwd):/data "$@". To 
understand the docker_run function read the section [Data, Tools and Computational Platforms (Docker)]</p>
<p style="font-size: 12px; padding-left: 30px"><b> staphb/fasttree:</b> is the docker image</p>
<p style="font-size: 12px; padding-left: 30px"><b> FastTree:</b> is the tool</p>
<p style="font-size: 12px; padding-left: 30px"><b> -nt:</b> specifies that the input alignment is 
nucleotide</p>
<p style="font-size: 12px; padding-left: 30px"><b> gtr:</b> specifies the evolutionary model</p>
<p style="font-size: 12px; padding-left: 30px"><b> clean.full.SNPs.aln:</b> input alignment</p>
<p style="font-size: 12px; padding-left: 30px"><b> clean.full.SNPs.aln.tree:</b> specifies the output tree 
name</p>

<p align="justify">Let’s explore output of our previous command using the command:</p>

<code><b>ls -lh clean.full.*</code></b>
<br>


<p align="center"><img width="400" height=“400” alt="logo"
  src="assets/images/logo/phylo_5.png"></p>

<p align="justify">Our maximum likelihood tree is labelled clean.full.SNPs.aln.tree. We can visualise this 
using figtree. Type the command: </p>

<code><b>figtree clean.full.SNPs.aln.tree &</code></b>
<br>

<p align="justify">Or you can simply right click on the on the <b>clean.full.SNPs.aln.tree</b> file and open it using figtree. How to download and 
install figtree has been described under "Data, Tools and Computational Platforms".</p>

<p align="justify">You will have the output:</p>

<p align="center"><img width="400" height=“400” alt="logo"
src="assets/images/logo/phylo_6.png"></p>

<h4>Interpreting a phylogenetic tree</h4>

<p align="justify">We need to remember that the tree we have created is from a SNP only alignment. Strain 
21127_1#30 sits on a long branch on its own; which means that it has unique mutations not found in the 
other strains. Remember that the longer the branch a strain has, the more mutations it has. Strain 
15608_3#13 and the reference strain are identical to each other, and have some mutations found also in 
strains 17150_4#79, 17175_6#87 and 13415_4#10. Strains 17150_4#79, 17175_6#87 and 13415_4#10, have some 
unique mutations.</p>

<br>
<h5>Identifying recombination using Gubbins</h5>

<p align=“justify”><I>You will continue using the variant calling output to learn and practice this section.</p></I>

<h4>Overview</h4>
<p align="justify">Many bacteria participate in high rates of homologous recombination. This means that 
they donate and receive segments of DNA from each other. In the context of a phylogenetic tree, in which 
we are comparing similar and dissimilar regions to determine relatedness of isolates this can prove 
problematic resulting in branch lengths which reflect recombinations rather than divergence. This is very 
important for <I>Streptococcus pneumoniae</I> which is naturally competent, meaning it can readily take up 
DNA. </p>

<p align="justify">Gubbins (Genealogies Unbiased By recomBinations In Nucleotide Sequences) is an 
algorithm that iteratively identifies loci containing elevated densities of base substitutions while 
concurrently constructing a phylogeny based on the putative point mutations outside of these regions. 
Simulations demonstrate the algorithm generates highly accurate reconstructions under realistic models of 
short-term diversification of sequences through both point mutation and recombination, and can be run on 
alignments of many hundreds of bacterial genome sequences. It is therefore not appropriate for looking at 
recombination across species-wide diversity - this can be done gene-by-gene using software such as 
fastGEAR. Instead, it works on samples of limited diversity, sharing a recent common ancestor - a strain 
or lineage.</p>

<p align="justify">The required input file for Gubbins is a whole genome FASTA alignment. Each sequence 
should have a unique identifier, and special characters should be avoided. The sequences should only use 
the characters ACGT (DNA bases),N (unknown base) or - (alignment gap). If a starting tree is to be 
included, then this should be a Newick format.The alignment is most easily generated through mapping 
sequences against a reference sequence. This can be achieved with the popular mapping software Snippy.</p>

<p align=”left”>Further reading: <a 
href="https://github.com/nickjcroucher/gubbins/blob/master/docs/gubbins_manual.md#analysing-whole-genome-alignments-with-gubbins">Analysing 
whole genome alignments with Gubbins</a></p>

<h4>Tool(s)</h4>
<p align="justify">For this section, we will be using Gubbins. You can download it from a docker 
repository using the command:</p>

<code><b>docker pull sangerpathogens/gubbins</code></b>
<br>

<p align="justify">You can view gubbins commands as follows:</p>

<code><b>docker run sangerpathogens/gubbins gubbins run_gubbins.py -h</code></b>
<br>

<p align="center"><img width="400" height=“400” alt="logo"
src="assets/images/logo/phylo_8.png"></p>

<h4>Identifying recombination</h4>
<p align="justify">We will run the gubbins tool on a full genome alignment and not SNPs alignment. Now 
let’s run this command:</p>

<code><b>docker_run sangerpathogens/gubbins run_gubbins.py --mar -p /data/output 
/data/clean.full.aln</code></b>
<br>

<p style="font-size: 12px; padding-left: 30px"><b>An explanation of these commands are as follows:</b></p>
<p style="font-size: 12px; padding-left: 30px"><b>docker_run:</b>is a function to start a container. The 
function includes the following flags: docker run --rm=True -u $(id -u):$(id -g) -v $(pwd):/data "$@". To 
understand the docker_run function read the section [Data and Computational Platforms (Docker)]</p>
<p style="font-size: 12px; padding-left: 30px"><b> sanger/gubbins:</b> is the docker image</p>
<p style="font-size: 12px; padding-left: 30px"><b> run_gubbins.py:</b> is the tool/program</p>
<p style="font-size: 12px; padding-left: 30px"><b> --mar:</b> use marginal ancestral reconstruction with 
RAxML instead of the default pyjar</p>
<p style="font-size: 12px; padding-left: 30px"><b> -p /data/output:</b> Output to /data dir in the 
container with a prefix of output. As /data dir is the mount point of $PWD of your host due to -v, you 
will see the files in the $PWD of your host</p>
<p style="font-size: 12px; padding-left: 30px"><b> /data/clean.full.aln:</b> input file</p>

<p align="justify">This command can take a few minutes to run.</p>

<p align="justify">Let's look at what gubbins has done using the command (ls -l output.*)</p>

<p style="font-size: 12px; padding-left: 30px"><b>An explanation of these commands are as follows:</b></p>
<p style="font-size: 12px; padding-left: 30px"><b>output.branch_base_reconstruction.embl:</b> Base 
substitution reconstruction in EMBL format</p>
<p style="font-size: 12px; padding-left: 30px"><b> output.recombination_predictions.embl:</b> 
Recombination predictions in EMBL file format.</p>
<p style="font-size: 12px; padding-left: 30px"><b> output.recombination_predictions.gff:</b> Recombination 
predictions in GFF format</p>
<p style="font-size: 12px; padding-left: 30px"><b> output.filtered_polymorphic_sites.phylip:</b> Phylip 
format alignment of filtered polymorphic sites used to generate the phylogeny in the final iteration</p>
<p style="font-size: 12px; padding-left: 30px"><b> output.final_tree.tre:</b> this file contains the final 
phylogeny in Newick format; branch lengths are in point mutations</p>
<p style="font-size: 12px; padding-left: 30px"><b> output.node_labelled.final_tree.tre:</b> final 
phylogenetic tree in Newick format but with internal node labels; branch lengths are in point 
mutations</p>
<p style="font-size: 12px; padding-left: 30px"><b> output.log:</b> log file specifying the software used 
at each step of the analysis, with accompanying citations</p>
<p style="font-size: 12px; padding-left: 30px"><b> output.per_branch_statistics.csv:</b> per branch 
reporting of the base substitutions inside and outside recombination events</p>
<p style="font-size: 12px; padding-left: 30px"><b> output.summary_of_snp_distribution.vcf:</b> VCF file 
summarising the distribution of point mutations</p>

<p align="justify">You can explore these output files using (head) command. For example 
“<b>output.recombination_predictions.gff</b>" is a gff file that contains a record of each recombination 
block identified, how many SNPs it contains, and what samples are affected.</p>

<p align="center"><img width="400" height="400" alt="logo"
src="assets/images/logo/phylo_9.png"></p>

<p align="justify"><b>output.final_tree.tre</b> is a phylogeny without recombination regions. Read in the 
gubbins filtered tree and plot with ggtree, or visualise in Figtree.</p>

<p align="center"><img width="400" height="400" alt="logo"
src="assets/images/logo/phylo_10.png"></p>

<p align="justify">FigTree output</p>

<p align="center"><img width="400" height="200" alt="logo"
src="assets/images/logo/phylo_11.png"></p>


<p align="justify">We can also visualise the recombination blocks using a web tool called phandango. Using 
your browser, navigate to:  <a>https://jameshadfield.github.io/phandango/#/</a></p>

<p align="justify">You will need the following files (drag and drop):
<ul style="padding-left: 50px">
<li>output.final_tree.tre</li>
<li>output.recombination_predictions.gff</li>
<li>PROKKA_12252022.gff (output from Prokka)</li>
</ul>
</p>

<p align="justify">Phandango should automatically display blocks of recombination in red (ancestral) and 
blue (specific to a sample)</p>


<h5>Clustering using popPUNK</h5>

<br>
<p align=“justify”><I>Before you being this section, navigate to the <a 
href="https://drive.google.com/drive/folders/1UlHvrhniA8KxxsVygqgOkkjFNsg8GIfk?usp=share_link">clustering</a> 
folder. You will use this folder
and its contents to learn and practice this section.</p></I>

<h4>Overview</h4>
<p align="justify">PopPUNK is a tool for clustering genomes. We refer to the clusters as 
variable-length-k-mer clusters, or <b>VLKCs</b>. Biologically, these clusters typically represent distinct 
strains. We refer to subclusters of strains as lineages.</p>

<p align="justify">The figure below shows an overview of how to run popPUNK</p>

<p align="center"><img width="500" height="700” alt="logo"
src="assets/images/logo/phylo_12.png"></p>

<p align="center"><a href="https://poppunk.readthedocs.io/en/latest/overview.html">PopPUNK 
documentation</a>

<h4>Tool</h4>
<p align="justify">You can download popPUNK from a docker repository using the command:</p>

<code><b>docker pull staphb/poppunk</code></b>
<br>

<h4>Clustering</h4>
<p align="justify"><b>Query assignment steps:</b></p>

<p align="justify"><b>1. Download database</b></p>

<p align="justify">As we are working on Streptococcus pneumoniae, we will download the <a 
href="https://gps-project.cog.sanger.ac.uk/GPS_v6.zip">GPS reference database</a> and <a 
href="https://www.pneumogen.net/gps/GPS_v6_external_clusters.csv">GPS designation</a>.
which we will use to cluster our genome. You can also access reference genomes of other bacterial species 
from this <a href="https://poppunk.readthedocs.io/en/latest/sketching.html">site</a>. </p>

<p align="justify">The Streptococcus pneumoniae GPS reference genome database is saved in you directory as 
(GPS_v6) and the GPS designation as (GPS_v6_external_clusters.csv) </p>

<p align="justify"><b>2. Create a text file with details of your samples</b></p>

<p align="justify">Create a file which lists your sample names and paths to their sequence data using the command:</p>

<b><code>echo -e "17150_4#79\t/data/17150_4#79/17150_4#79_1.fastq.gz\t/data/17150_4#79/17150_4#79_2.fastq.gz" > 
poppunk_input.tsv</code></b>

<p align="justify">This text file contains the sample names and their sequence data. It has no header, is tab separated, and contains the sample 
name in the first column. Subsequent columns may contain paths to either assembled or raw read data (the type will automatically be inferred by 
checking for the presence of quality scores). If you <code><b>cat poppunk_input.tsv</code></b></p>

<p align="justify">You will have the output</p>


<p align="center"><img width="400" height="240" alt="logo"
  src="assets/images/logo/phylo_14.png"></p>


<p align="justify"><b>3. Clustering your genomes</b></p>

<p align="justify">The command to cluster your genomes is as follows:</p>

<code><b>docker_run staphb/poppunk poppunk_assign --db GPS_v6 --query /data/poppunk_input.tsv --output poppunk_clusters 
--external-clustering GPS_v6_external_clusters.csv</code></b>
<br>

<p style="font-size: 12px; padding-left: 30px"><b>An explanation of these commands are as follows:</b></p>
<p style="font-size: 12px; padding-left: 30px"><b>docker_run:</b> is a function to start a container. The 
function includes the following flags: docker run --rm=True -u $(id -u):$(id -g) -v $(pwd):/data "$@". To 
understand the docker_run function read the section [Data, Tools and Computational Platforms (Docker)]</p>
<p style="font-size: 12px; padding-left: 30px"><b> staphb/poppunk:</b> is the docker image</p>
<p style="font-size: 12px; padding-left: 30px"><b> poppunk_assign:</b> is the tool/program/script</p>
<p style="font-size: 12px; padding-left: 30px"><b>--query /data/poppunk_input.tsv:</b> receives input from the 
list.txt</p>
<p style="font-size: 12px; padding-left: 30px"><b>--output /data/poppunk_clusters:</b> specifies the 
output file </p>
<p style="font-size: 12px; padding-left: 30px"><b>--external-clustering 
/data/GPS_v6_external_clusters.csv:</b> directory containing the GPS clusters for the references</p>
<p style="font-size: 12px; padding-left: 30px"><b>--threads 8:</b> specifies virtual cores</p>

<p align="justify">Upon completion, a new folder "<b>poppunk_clusters</b>" will be generated. Navigate to 
this folder and explore its content.</p>

<p align="justify">The output files: 
<p style="font-size: 12px; padding-left: 30px"><b>poppunk_clusters_clusters.csv:</b> popPUNK clusters with 
dataset specific nomenclature</p>
<p style="font-size: 12px; padding-left: 30px"><b>poppunk_clusters_external_clusters.csv:</b> GPSC v6 
scheme designations</p>

<p align="justify">We can explore "poppunk_clusters_clusters.csv"</p>

<p align="justify"><b>Note:</b>
<ul style="padding-left: 50px">
<li>If a strain has already been assigned a cluster, please rename to run popPUNK (this is to avoid 
crushing the tool)</li>
<li>Novel Clusters are assigned NA in the _external_clusters.csv as they have not been defined in the v6 
dataset used to designate the GPSCs. Please email: globalpneumoseq@gmail.com to have novel clusters added 
to the database and a GPSC cluster name assigned after you have checked for low level contamination which 
may contribute to biased accessory distances.</li>
</ul>
</p>

<p align="justify">We can explore "poppunk_clusters_clusters.csv"</p>
  
 </div>
    

<script>
/* Loop through all dropdown buttons to toggle between hiding and showing its dropdown content - This allows the user to have multiple dropdowns without any conflict */
var dropdown = document.getElementsByClassName("dropdown-btn");
var i;
    
for (i = 0; i < dropdown.length; i++) {
  dropdown[i].addEventListener("click", function() {  
    this.classList.toggle("active");
    var dropdownContent = this.nextElementSibling;
    if (dropdownContent.style.display === "block") {  

      dropdownContent.style.display = "none";
    } else {
      dropdownContent.style.display = "block";
    }
  });
} 
</script>

</body>
</html>

